<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 5 Count data | The Earth Hologenome Initiative Data Analysis Workflow</title>
  <meta name="description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 5 Count data | The Earth Hologenome Initiative Data Analysis Workflow" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />
  <meta property="og:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="github-repo" content="earthhologenome/EHI_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Count data | The Earth Hologenome Initiative Data Analysis Workflow" />
  
  <meta name="twitter:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="twitter:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />

<meta name="author" content="Ostaizka Aizpurua" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2023-11-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="sequencing-assessment.html"/>
<link rel="next" href="taxonomic-composition.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><img src="images/ehi_logo.png" alt="Earth Hologenome Initiative"></li>
<li align="center"><b>ANALYSIS WORFLOW</b></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#download-mock-data"><i class="fa fa-check"></i><b>1.2</b> Download mock data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-summary.html"><a href="data-summary.html"><i class="fa fa-check"></i><b>2</b> Data summary</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-summary.html"><a href="data-summary.html#load-the-data"><i class="fa fa-check"></i><b>2.1</b> Load the data</a></li>
<li class="chapter" data-level="2.2" data-path="data-summary.html"><a href="data-summary.html#general-data-statistics"><i class="fa fa-check"></i><b>2.2</b> General data statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-samples"><i class="fa fa-check"></i>Number of samples</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-discarded-data"><i class="fa fa-check"></i>Amount of discarded data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-host-data"><i class="fa fa-check"></i>Amount of host data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-metagenomic-data"><i class="fa fa-check"></i>Amount of metagenomic data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-estimated-prokaryotic"><i class="fa fa-check"></i>Amount of estimated prokaryotic</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-summary.html"><a href="data-summary.html#general-mag-statistics"><i class="fa fa-check"></i><b>2.3</b> General MAG statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags"><i class="fa fa-check"></i>Number of MAGs</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags-without-species-level-annotation"><i class="fa fa-check"></i>Number of MAGs without species-level annotation</a></li>
<li class="chapter" data-level="2.3.1" data-path="data-summary.html"><a href="data-summary.html#number-of-phylums"><i class="fa fa-check"></i><b>2.3.1</b> Number of phylums</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-summary.html"><a href="data-summary.html#geographic-distribution"><i class="fa fa-check"></i><b>2.4</b> Geographic distribution</a></li>
<li class="chapter" data-level="2.5" data-path="data-summary.html"><a href="data-summary.html#prepare-the-colour-layout"><i class="fa fa-check"></i><b>2.5</b> Prepare the colour layout</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>3</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="3.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-phylogeny"><i class="fa fa-check"></i><b>3.1</b> MAG phylogeny</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-mag-metadata"><i class="fa fa-check"></i>Prepare MAG metadata</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylogenetic-tree"><i class="fa fa-check"></i>Plot phylogenetic tree</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylum-color-legend"><i class="fa fa-check"></i>Plot phylum color legend</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-quality"><i class="fa fa-check"></i><b>3.2</b> MAG quality</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-input-table"><i class="fa fa-check"></i>Prepare input table</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-biplot-chart"><i class="fa fa-check"></i>Create biplot chart</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-boxplot-charts"><i class="fa fa-check"></i>Create boxplot charts</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-composite-figure"><i class="fa fa-check"></i>Create composite figure</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-attributes-of-mags"><i class="fa fa-check"></i><b>3.3</b> Functional attributes of MAGs</a></li>
<li class="chapter" data-level="3.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-ordination-of-mags"><i class="fa fa-check"></i><b>3.4</b> Functional ordination of MAGs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html"><i class="fa fa-check"></i><b>4</b> Sequencing Assessment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#dna-fractions"><i class="fa fa-check"></i><b>4.1</b> DNA fractions</a></li>
<li class="chapter" data-level="4.2" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#estimated-vs.-mapped-prokaryotic-fraction"><i class="fa fa-check"></i><b>4.2</b> Estimated vs.Â mapped prokaryotic fraction</a></li>
<li class="chapter" data-level="4.3" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#additional-sequencing-needed"><i class="fa fa-check"></i><b>4.3</b> Additional sequencing needed</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="count-data.html"><a href="count-data.html"><i class="fa fa-check"></i><b>5</b> Count data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-data.html"><a href="count-data.html#minimum-coverage-filtering"><i class="fa fa-check"></i><b>5.1</b> Minimum coverage filtering</a></li>
<li class="chapter" data-level="5.2" data-path="count-data.html"><a href="count-data.html#genome-size-normalisation"><i class="fa fa-check"></i><b>5.2</b> Genome size normalisation</a></li>
<li class="chapter" data-level="5.3" data-path="count-data.html"><a href="count-data.html#count-table"><i class="fa fa-check"></i><b>5.3</b> Count table</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html"><i class="fa fa-check"></i><b>6</b> Taxonomic composition</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html#taxonomy-stacked-barplot"><i class="fa fa-check"></i><b>6.1</b> Taxonomy stacked barplot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> Diversity analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#data-preparation"><i class="fa fa-check"></i><b>7.1</b> Data preparation</a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity"><i class="fa fa-check"></i><b>7.2</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-metrics"><i class="fa fa-check"></i>Alpha diversity metrics</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-plots"><i class="fa fa-check"></i>Alpha diversity plots</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-comparisons"><i class="fa fa-check"></i>Alpha diversity comparisons</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#relationship-between-alpha-diversity-and-sequencing-effort"><i class="fa fa-check"></i>Relationship between alpha diversity and sequencing effort</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity"><i class="fa fa-check"></i><b>7.3</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-test"><i class="fa fa-check"></i>Beta diversity test</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-plot"><i class="fa fa-check"></i>Beta diversity plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://www.earthhologenome.org" target="blank"><b>Earth Hologenome Initiative</b></a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Earth Hologenome Initiative Data Analysis Workflow</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="count-data" class="section level1 hasAnchor" number="5">
<h1><span class="header-section-number">Chapter 5</span> Count data<a href="count-data.html#count-data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This workflow transforms and visualises the quantitative information of the MAGs generated through the EHI pipeline. Note the workflow relies on previous scripts, so make sure you go through the general pipeline before starting working on this document.</p>
<div id="minimum-coverage-filtering" class="section level2 hasAnchor" number="5.1">
<h2><span class="header-section-number">5.1</span> Minimum coverage filtering<a href="count-data.html#minimum-coverage-filtering" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Mapping of sequencing reads against the reference genome catalogue is not perfect, and in consequence, all MAGs tend to get a few reads assigned. Implementing a minimum genome coverage filter aims at minimising artificial inflation of diversity due to this artefact of genome-resolved metagenomic analysis. The EHI pipeline implements a default coverage threshold of 0.3 (%30) of a genome. However, if the sequencing depth is low and uneven across samples, this filtering can also introduce distortion, by introducing false positives.</p>
<div class="sourceCode" id="cb46"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb46-1"><a href="count-data.html#cb46-1" tabindex="-1"></a>min_coverage<span class="ot">=</span><span class="fl">0.3</span></span>
<span id="cb46-2"><a href="count-data.html#cb46-2" tabindex="-1"></a>count_table_cov <span class="ot">&lt;-</span> coverage_table <span class="sc">%&gt;%</span></span>
<span id="cb46-3"><a href="count-data.html#cb46-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="fu">across</span>(<span class="fu">everything</span>(), <span class="sc">~</span> <span class="fu">ifelse</span>(. <span class="sc">&gt;</span> min_coverage, <span class="dv">1</span>, <span class="dv">0</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb46-4"><a href="count-data.html#cb46-4" tabindex="-1"></a>  <span class="fu">map2_df</span>(., count_table, <span class="sc">~</span> .x <span class="sc">*</span> .y) <span class="sc">%&gt;%</span></span>
<span id="cb46-5"><a href="count-data.html#cb46-5" tabindex="-1"></a>  <span class="fu">as.data.frame</span>()</span>
<span id="cb46-6"><a href="count-data.html#cb46-6" tabindex="-1"></a><span class="fu">rownames</span>(count_table_cov) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(coverage_table)</span></code></pre></div>
</div>
<div id="genome-size-normalisation" class="section level2 hasAnchor" number="5.2">
<h2><span class="header-section-number">5.2</span> Genome size normalisation<a href="count-data.html#genome-size-normalisation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Bacterial genomes can vary between 1 and 8 MB, which make relative representation of each genome dependent on its size. To account for genome size biases, read-counts can be normalised by applying a normalisation factor that modifies the read numbers according to the size of each genome compared to the average genome size in the dataset.</p>
<div class="sourceCode" id="cb47"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb47-1"><a href="count-data.html#cb47-1" tabindex="-1"></a>genome_read_sizes <span class="ot">&lt;-</span> mags_table[<span class="fu">rownames</span>(count_table_cov),] <span class="sc">%&gt;%</span></span>
<span id="cb47-2"><a href="count-data.html#cb47-2" tabindex="-1"></a>    <span class="fu">select</span>(mag_size) <span class="sc">%&gt;%</span></span>
<span id="cb47-3"><a href="count-data.html#cb47-3" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">mag_size =</span> mag_size <span class="sc">/</span> <span class="dv">150</span>) <span class="sc">%&gt;%</span></span>
<span id="cb47-4"><a href="count-data.html#cb47-4" tabindex="-1"></a>    <span class="fu">pull</span>()</span>
<span id="cb47-5"><a href="count-data.html#cb47-5" tabindex="-1"></a>count_table_cov_size <span class="ot">&lt;-</span> <span class="fu">sweep</span>(count_table_cov, <span class="dv">1</span>, genome_read_sizes, <span class="st">&quot;/&quot;</span>)</span></code></pre></div>
</div>
<div id="count-table" class="section level2 hasAnchor" number="5.3">
<h2><span class="header-section-number">5.3</span> Count table<a href="count-data.html#count-table" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Once low-coverage genome counts have been filtered out, and the read counts have been normalised into genome counts, we can visualise the relative MAG abundances per sample. Note that the script shown below generates a count table in which the scale is log-transformed.</p>
<div class="sourceCode" id="cb48"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb48-1"><a href="count-data.html#cb48-1" tabindex="-1"></a>vertical_tree <span class="ot">&lt;-</span> <span class="fu">force.ultrametric</span>(tree,<span class="at">method=</span><span class="st">&quot;extend&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb48-2"><a href="count-data.html#cb48-2" tabindex="-1"></a>        <span class="fu">ggtree</span>(., <span class="at">size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<pre class="script-output"><code>***************************************************************
*                          Note:                              *
*    force.ultrametric does not include a formal method to    *
*    ultrametricize a tree &amp; should only be used to coerce    *
*   a phylogeny that fails is.ultramtric due to rounding --   *
*    not as a substitute for formal rate-smoothing methods.   *
***************************************************************</code></pre>
<div class="sourceCode" id="cb50"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb50-1"><a href="count-data.html#cb50-1" tabindex="-1"></a><span class="co">#Add phylum colors</span></span>
<span id="cb50-2"><a href="count-data.html#cb50-2" tabindex="-1"></a>vertical_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(vertical_tree, heatmap, <span class="at">offset=</span><span class="dv">0</span>, <span class="at">width=</span><span class="fl">0.1</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb50-3"><a href="count-data.html#cb50-3" tabindex="-1"></a>    <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>colors_alphabetic)</span>
<span id="cb50-4"><a href="count-data.html#cb50-4" tabindex="-1"></a></span>
<span id="cb50-5"><a href="count-data.html#cb50-5" tabindex="-1"></a><span class="co">#Reset fill scale</span></span>
<span id="cb50-6"><a href="count-data.html#cb50-6" tabindex="-1"></a>vertical_tree <span class="ot">&lt;-</span> vertical_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb50-7"><a href="count-data.html#cb50-7" tabindex="-1"></a></span>
<span id="cb50-8"><a href="count-data.html#cb50-8" tabindex="-1"></a><span class="co">#Add counts</span></span>
<span id="cb50-9"><a href="count-data.html#cb50-9" tabindex="-1"></a>vertical_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(vertical_tree, <span class="fu">log10</span>(count_table_cov_size), <span class="at">offset=</span><span class="fl">0.04</span>, <span class="at">width=</span><span class="fl">3.5</span>, <span class="at">colnames=</span><span class="cn">TRUE</span>, <span class="at">colnames_angle=</span><span class="dv">90</span>, <span class="at">font.size=</span><span class="dv">2</span>, <span class="at">colnames_position=</span><span class="st">&quot;top&quot;</span>, <span class="at">colnames_offset_y =</span> <span class="dv">9</span>) <span class="sc">+</span></span>
<span id="cb50-10"><a href="count-data.html#cb50-10" tabindex="-1"></a>    <span class="fu">vexpand</span>(.<span class="dv">08</span>) <span class="sc">+</span></span>
<span id="cb50-11"><a href="count-data.html#cb50-11" tabindex="-1"></a>    <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">&quot;off&quot;</span>) <span class="sc">+</span></span>
<span id="cb50-12"><a href="count-data.html#cb50-12" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb50-13"><a href="count-data.html#cb50-13" tabindex="-1"></a></span>
<span id="cb50-14"><a href="count-data.html#cb50-14" tabindex="-1"></a><span class="co">#Plot tree</span></span>
<span id="cb50-15"><a href="count-data.html#cb50-15" tabindex="-1"></a>vertical_tree <span class="sc">+</span></span>
<span id="cb50-16"><a href="count-data.html#cb50-16" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&#39;none&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/count_table-1.png" width="768" /></p>
<p>We can then plot a legend for the count data.</p>
<div class="sourceCode" id="cb51"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb51-1"><a href="count-data.html#cb51-1" tabindex="-1"></a><span class="co">#Add count scale</span></span>
<span id="cb51-2"><a href="count-data.html#cb51-2" tabindex="-1"></a>countscale <span class="ot">&lt;-</span> <span class="fu">seq</span>(<span class="fu">log</span>(<span class="fu">max</span>(count_table_cov_size)), <span class="dv">0</span>, <span class="at">length.out =</span> <span class="dv">5</span>)</span>
<span id="cb51-3"><a href="count-data.html#cb51-3" tabindex="-1"></a></span>
<span id="cb51-4"><a href="count-data.html#cb51-4" tabindex="-1"></a>count_legend <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">value =</span> countscale, <span class="at">x =</span> <span class="fu">c</span>(<span class="fu">round</span>(<span class="fu">exp</span>(countscale[<span class="dv">1</span>]),<span class="dv">2</span>),<span class="fu">round</span>(<span class="fu">exp</span>(countscale[<span class="dv">2</span>]),<span class="dv">2</span>),<span class="fu">round</span>(<span class="fu">exp</span>(countscale[<span class="dv">3</span>]),<span class="dv">2</span>),<span class="fu">round</span>(<span class="fu">exp</span>(countscale[<span class="dv">4</span>]),<span class="dv">2</span>),<span class="st">&quot;0&quot;</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb51-5"><a href="count-data.html#cb51-5" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">x =</span> <span class="fu">factor</span>(x, <span class="at">levels =</span> x)) <span class="sc">%&gt;%</span></span>
<span id="cb51-6"><a href="count-data.html#cb51-6" tabindex="-1"></a>    <span class="fu">ggplot</span>(., <span class="fu">aes</span>(<span class="at">x =</span> x, <span class="at">y =</span> <span class="fl">0.2</span>)) <span class="sc">+</span></span>
<span id="cb51-7"><a href="count-data.html#cb51-7" tabindex="-1"></a>  <span class="fu">geom_tile</span>(<span class="fu">aes</span>(<span class="at">fill =</span> value, <span class="at">y =</span> <span class="fl">0.2</span>), <span class="at">color=</span><span class="st">&quot;#CCCCCC&quot;</span>) <span class="sc">+</span></span>
<span id="cb51-8"><a href="count-data.html#cb51-8" tabindex="-1"></a>    <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;white&quot;</span>, <span class="at">high =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>) <span class="sc">+</span></span>
<span id="cb51-9"><a href="count-data.html#cb51-9" tabindex="-1"></a>  <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb51-10"><a href="count-data.html#cb51-10" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&#39;none&#39;</span>,</span>
<span id="cb51-11"><a href="count-data.html#cb51-11" tabindex="-1"></a>            <span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">90</span>, <span class="at">vjust =</span> <span class="dv">1</span>, <span class="at">hjust=</span><span class="dv">1</span>, <span class="at">size=</span><span class="dv">8</span>))</span>
<span id="cb51-12"><a href="count-data.html#cb51-12" tabindex="-1"></a></span>
<span id="cb51-13"><a href="count-data.html#cb51-13" tabindex="-1"></a><span class="co">#Arrange both legends</span></span>
<span id="cb51-14"><a href="count-data.html#cb51-14" tabindex="-1"></a><span class="fu">grid.arrange</span>(<span class="at">grobs =</span> <span class="fu">list</span>(phyla_legend,count_legend), <span class="at">layout_matrix =</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">2</span>),<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">3</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/count_table_legend-1.png" width="768" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="sequencing-assessment.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="taxonomic-composition.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
