<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 2 Data summary | The Earth Hologenome Initiative Data Analysis Workflow</title>
  <meta name="description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 2 Data summary | The Earth Hologenome Initiative Data Analysis Workflow" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />
  <meta property="og:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="github-repo" content="earthhologenome/EHI_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 2 Data summary | The Earth Hologenome Initiative Data Analysis Workflow" />
  
  <meta name="twitter:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="twitter:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />

<meta name="author" content="Ostaizka Aizpurua" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2023-11-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="mag-catalogue.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><img src="images/ehi_logo.png" alt="Earth Hologenome Initiative"></li>
<li align="center"><b>ANALYSIS WORFLOW</b></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#download-mock-data"><i class="fa fa-check"></i><b>1.2</b> Download mock data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-summary.html"><a href="data-summary.html"><i class="fa fa-check"></i><b>2</b> Data summary</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-summary.html"><a href="data-summary.html#load-the-data"><i class="fa fa-check"></i><b>2.1</b> Load the data</a></li>
<li class="chapter" data-level="2.2" data-path="data-summary.html"><a href="data-summary.html#general-data-statistics"><i class="fa fa-check"></i><b>2.2</b> General data statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-samples"><i class="fa fa-check"></i>Number of samples</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-discarded-data"><i class="fa fa-check"></i>Amount of discarded data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-host-data"><i class="fa fa-check"></i>Amount of host data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-metagenomic-data"><i class="fa fa-check"></i>Amount of metagenomic data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-estimated-prokaryotic"><i class="fa fa-check"></i>Amount of estimated prokaryotic</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-summary.html"><a href="data-summary.html#general-mag-statistics"><i class="fa fa-check"></i><b>2.3</b> General MAG statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags"><i class="fa fa-check"></i>Number of MAGs</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags-without-species-level-annotation"><i class="fa fa-check"></i>Number of MAGs without species-level annotation</a></li>
<li class="chapter" data-level="2.3.1" data-path="data-summary.html"><a href="data-summary.html#number-of-phylums"><i class="fa fa-check"></i><b>2.3.1</b> Number of phylums</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-summary.html"><a href="data-summary.html#geographic-distribution"><i class="fa fa-check"></i><b>2.4</b> Geographic distribution</a></li>
<li class="chapter" data-level="2.5" data-path="data-summary.html"><a href="data-summary.html#prepare-the-colour-layout"><i class="fa fa-check"></i><b>2.5</b> Prepare the colour layout</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>3</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="3.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-phylogeny"><i class="fa fa-check"></i><b>3.1</b> MAG phylogeny</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-mag-metadata"><i class="fa fa-check"></i>Prepare MAG metadata</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylogenetic-tree"><i class="fa fa-check"></i>Plot phylogenetic tree</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylum-color-legend"><i class="fa fa-check"></i>Plot phylum color legend</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-quality"><i class="fa fa-check"></i><b>3.2</b> MAG quality</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-input-table"><i class="fa fa-check"></i>Prepare input table</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-biplot-chart"><i class="fa fa-check"></i>Create biplot chart</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-boxplot-charts"><i class="fa fa-check"></i>Create boxplot charts</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-composite-figure"><i class="fa fa-check"></i>Create composite figure</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-attributes-of-mags"><i class="fa fa-check"></i><b>3.3</b> Functional attributes of MAGs</a></li>
<li class="chapter" data-level="3.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-ordination-of-mags"><i class="fa fa-check"></i><b>3.4</b> Functional ordination of MAGs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html"><i class="fa fa-check"></i><b>4</b> Sequencing Assessment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#dna-fractions"><i class="fa fa-check"></i><b>4.1</b> DNA fractions</a></li>
<li class="chapter" data-level="4.2" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#estimated-vs.-mapped-prokaryotic-fraction"><i class="fa fa-check"></i><b>4.2</b> Estimated vs. mapped prokaryotic fraction</a></li>
<li class="chapter" data-level="4.3" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#additional-sequencing-needed"><i class="fa fa-check"></i><b>4.3</b> Additional sequencing needed</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="count-data.html"><a href="count-data.html"><i class="fa fa-check"></i><b>5</b> Count data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-data.html"><a href="count-data.html#minimum-coverage-filtering"><i class="fa fa-check"></i><b>5.1</b> Minimum coverage filtering</a></li>
<li class="chapter" data-level="5.2" data-path="count-data.html"><a href="count-data.html#genome-size-normalisation"><i class="fa fa-check"></i><b>5.2</b> Genome size normalisation</a></li>
<li class="chapter" data-level="5.3" data-path="count-data.html"><a href="count-data.html#count-table"><i class="fa fa-check"></i><b>5.3</b> Count table</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html"><i class="fa fa-check"></i><b>6</b> Taxonomic composition</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html#taxonomy-stacked-barplot"><i class="fa fa-check"></i><b>6.1</b> Taxonomy stacked barplot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> Diversity analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#data-preparation"><i class="fa fa-check"></i><b>7.1</b> Data preparation</a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity"><i class="fa fa-check"></i><b>7.2</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-metrics"><i class="fa fa-check"></i>Alpha diversity metrics</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-plots"><i class="fa fa-check"></i>Alpha diversity plots</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-comparisons"><i class="fa fa-check"></i>Alpha diversity comparisons</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#relationship-between-alpha-diversity-and-sequencing-effort"><i class="fa fa-check"></i>Relationship between alpha diversity and sequencing effort</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity"><i class="fa fa-check"></i><b>7.3</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-test"><i class="fa fa-check"></i>Beta diversity test</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-plot"><i class="fa fa-check"></i>Beta diversity plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://www.earthhologenome.org" target="blank"><b>Earth Hologenome Initiative</b></a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Earth Hologenome Initiative Data Analysis Workflow</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data-summary" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">Chapter 2</span> Data summary<a href="data-summary.html#data-summary" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<div id="load-the-data" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Load the data<a href="data-summary.html#load-the-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The EHI pipeline outputs six data files in analysis batches. Each batch is assigned a batch code in the form of DMB1234. DMB stands for Dereplication &amp; Mapping Batch, which is the final step of the EHI pipeline. The six output files are:</p>
<p><strong>Microbiome count table</strong> ([BATCH]_counts.tsv.gz): this is a contingency table containing the number of sequencing reads from each sample mapped against each MAG, with MAG identifiers in rows and sample identifiers in columns.</p>
<p><strong>Microbiome coverage table</strong> ([BATCH]_coverage.tsv.gz): it has the same structure as the count table, but contains breadth-of-coverage information of each MAG in each sample. In other words, it contains the fraction of the genomes covered by mapped reads in each sample.</p>
<p><strong>Sample metadata table</strong> ([BATCH]_metadata.tsv.gz): it contains relevant metadata of the samples, including geographic origin, host species, sample type and statistics of sample preprocessing.</p>
<p><strong>Microbiome metadata table</strong> ([BATCH]_mag_info.tsv.gz): it contains relevant metadata of the MAGs, including taxonomy, genome completeness, contamination/redundancy and other quality metrics.</p>
<p><strong>Microbiome phylogenetic tree</strong> ([BATCH].tree.gz): this is the phylogenetic tree of the MAGs derived from the GTDB master tree after pruning all reference genomes. This file is used for phylogenetic analyses.</p>
<p><strong>Microbiome functional attribute table</strong> ([BATCH]_merged_kegg.tsv.gz): this is a contingency table containing the fullness levels of hundreds of KEGG modules, with MAG identifiers in rows and KEGG module identifiers in columns. This table is used for functional analyses.</p>
<p>You first need to download these documents from the server, using the links in the EHI data report. In this workflow, we assume the files are downloaded to the folder “data” in your working environment.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb5-1"><a href="data-summary.html#cb5-1" tabindex="-1"></a><span class="co"># Batch</span></span>
<span id="cb5-2"><a href="data-summary.html#cb5-2" tabindex="-1"></a>batch<span class="ot">=</span><span class="st">&quot;DMB0038&quot;</span></span>
<span id="cb5-3"><a href="data-summary.html#cb5-3" tabindex="-1"></a></span>
<span id="cb5-4"><a href="data-summary.html#cb5-4" tabindex="-1"></a><span class="co"># Microbiome count table</span></span>
<span id="cb5-5"><a href="data-summary.html#cb5-5" tabindex="-1"></a>count_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb5-6"><a href="data-summary.html#cb5-6" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;_counts.tsv.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>),</span>
<span id="cb5-7"><a href="data-summary.html#cb5-7" tabindex="-1"></a>      <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T)</span>
<span id="cb5-8"><a href="data-summary.html#cb5-8" tabindex="-1"></a></span>
<span id="cb5-9"><a href="data-summary.html#cb5-9" tabindex="-1"></a><span class="co"># Microbiome coverage table</span></span>
<span id="cb5-10"><a href="data-summary.html#cb5-10" tabindex="-1"></a>coverage_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb5-11"><a href="data-summary.html#cb5-11" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;_coverage.tsv.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>),</span>
<span id="cb5-12"><a href="data-summary.html#cb5-12" tabindex="-1"></a>      <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">row.names=</span><span class="dv">1</span>,<span class="at">header=</span>T)</span>
<span id="cb5-13"><a href="data-summary.html#cb5-13" tabindex="-1"></a></span>
<span id="cb5-14"><a href="data-summary.html#cb5-14" tabindex="-1"></a><span class="co"># Sample metadata table</span></span>
<span id="cb5-15"><a href="data-summary.html#cb5-15" tabindex="-1"></a>sample_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb5-16"><a href="data-summary.html#cb5-16" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;_metadata.tsv.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>),</span>
<span id="cb5-17"><a href="data-summary.html#cb5-17" tabindex="-1"></a>      <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>T) <span class="sc">%&gt;%</span></span>
<span id="cb5-18"><a href="data-summary.html#cb5-18" tabindex="-1"></a>      <span class="fu">rename</span>(<span class="at">sample=</span>EHI_plaintext) <span class="co"># rename column</span></span>
<span id="cb5-19"><a href="data-summary.html#cb5-19" tabindex="-1"></a></span>
<span id="cb5-20"><a href="data-summary.html#cb5-20" tabindex="-1"></a><span class="co"># Microbiome metadata table</span></span>
<span id="cb5-21"><a href="data-summary.html#cb5-21" tabindex="-1"></a>mags_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb5-22"><a href="data-summary.html#cb5-22" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;_mag_info.tsv.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>),</span>
<span id="cb5-23"><a href="data-summary.html#cb5-23" tabindex="-1"></a>      <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>T)</span>
<span id="cb5-24"><a href="data-summary.html#cb5-24" tabindex="-1"></a><span class="fu">rownames</span>(mags_table) <span class="ot">&lt;-</span> mags_table[,<span class="dv">1</span>] <span class="co"># add row names</span></span>
<span id="cb5-25"><a href="data-summary.html#cb5-25" tabindex="-1"></a></span>
<span id="cb5-26"><a href="data-summary.html#cb5-26" tabindex="-1"></a><span class="co"># Microbiome phylogenetic tree</span></span>
<span id="cb5-27"><a href="data-summary.html#cb5-27" tabindex="-1"></a>tree <span class="ot">&lt;-</span> <span class="fu">read.tree</span>(</span>
<span id="cb5-28"><a href="data-summary.html#cb5-28" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;.tree.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>))</span>
<span id="cb5-29"><a href="data-summary.html#cb5-29" tabindex="-1"></a></span>
<span id="cb5-30"><a href="data-summary.html#cb5-30" tabindex="-1"></a><span class="co"># Microbiome functional attribute table</span></span>
<span id="cb5-31"><a href="data-summary.html#cb5-31" tabindex="-1"></a>kegg_table <span class="ot">&lt;-</span> <span class="fu">read.table</span>(</span>
<span id="cb5-32"><a href="data-summary.html#cb5-32" tabindex="-1"></a>      <span class="fu">gunzip</span>(<span class="fu">paste0</span>(<span class="st">&quot;data/&quot;</span>,batch,<span class="st">&quot;_merged_kegg.tsv.gz&quot;</span>), <span class="at">remove=</span><span class="cn">FALSE</span>, <span class="at">overwrite=</span><span class="cn">TRUE</span>),</span>
<span id="cb5-33"><a href="data-summary.html#cb5-33" tabindex="-1"></a>      <span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>T, <span class="at">row.names=</span><span class="dv">1</span>)</span></code></pre></div>
</div>
<div id="general-data-statistics" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> General data statistics<a href="data-summary.html#general-data-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can then generate some general statistics to obtain an overview of your data.</p>
<div id="number-of-samples" class="section level3 unnumbered hasAnchor">
<h3>Number of samples<a href="data-summary.html#number-of-samples" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb6"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb6-1"><a href="data-summary.html#cb6-1" tabindex="-1"></a><span class="fu">ncol</span>(count_table)</span></code></pre></div>
<pre class="script-output"><code>[1] 38</code></pre>
</div>
<div id="amount-of-discarded-data" class="section level3 unnumbered hasAnchor">
<h3>Amount of discarded data<a href="data-summary.html#amount-of-discarded-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The value is in GBs (gigabases)</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb8-1"><a href="data-summary.html#cb8-1" tabindex="-1"></a><span class="fu">sum</span>(<span class="fu">round</span>(((sample_table<span class="sc">$</span>metagenomic_bases<span class="sc">+</span>sample_table<span class="sc">$</span>host_bases)<span class="sc">/</span>(<span class="dv">1</span><span class="sc">-</span>sample_table<span class="sc">$</span>bases_lost_fastp_percent))<span class="sc">-</span>(sample_table<span class="sc">$</span>metagenomic_bases<span class="sc">+</span>sample_table<span class="sc">$</span>host_bases)))<span class="sc">/</span><span class="dv">1000000000</span></span></code></pre></div>
<pre class="script-output"><code>[1] 6.410973</code></pre>
</div>
<div id="amount-of-host-data" class="section level3 unnumbered hasAnchor">
<h3>Amount of host data<a href="data-summary.html#amount-of-host-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The value is in GBs (gigabases)</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb10-1"><a href="data-summary.html#cb10-1" tabindex="-1"></a><span class="fu">sum</span>(sample_table<span class="sc">$</span>host_bases)<span class="sc">/</span><span class="dv">1000000000</span></span></code></pre></div>
<pre class="script-output"><code>[1] 7.046948</code></pre>
</div>
<div id="amount-of-metagenomic-data" class="section level3 unnumbered hasAnchor">
<h3>Amount of metagenomic data<a href="data-summary.html#amount-of-metagenomic-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The value is in GBs (gigabases)</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb12-1"><a href="data-summary.html#cb12-1" tabindex="-1"></a><span class="fu">sum</span>(sample_table<span class="sc">$</span>metagenomic_bases)<span class="sc">/</span><span class="dv">1000000000</span></span></code></pre></div>
<pre class="script-output"><code>[1] 133.1334</code></pre>
</div>
<div id="amount-of-estimated-prokaryotic" class="section level3 unnumbered hasAnchor">
<h3>Amount of estimated prokaryotic<a href="data-summary.html#amount-of-estimated-prokaryotic" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb14"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb14-1"><a href="data-summary.html#cb14-1" tabindex="-1"></a><span class="fu">sum</span>(sample_table<span class="sc">$</span>metagenomic_bases <span class="sc">*</span> sample_table<span class="sc">$</span>singlem_fraction)<span class="sc">/</span><span class="dv">1000000000</span></span></code></pre></div>
<pre class="script-output"><code>[1] 88.89506</code></pre>
</div>
</div>
<div id="general-mag-statistics" class="section level2 hasAnchor" number="2.3">
<h2><span class="header-section-number">2.3</span> General MAG statistics<a href="data-summary.html#general-mag-statistics" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div id="number-of-mags" class="section level3 unnumbered hasAnchor">
<h3>Number of MAGs<a href="data-summary.html#number-of-mags" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb16"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb16-1"><a href="data-summary.html#cb16-1" tabindex="-1"></a><span class="fu">nrow</span>(count_table)</span></code></pre></div>
<pre class="script-output"><code>[1] 581</code></pre>
</div>
<div id="number-of-mags-without-species-level-annotation" class="section level3 unnumbered hasAnchor">
<h3>Number of MAGs without species-level annotation<a href="data-summary.html#number-of-mags-without-species-level-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>These are the MAGs that could be considered “new” species</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb18-1"><a href="data-summary.html#cb18-1" tabindex="-1"></a>mags_table <span class="sc">%&gt;%</span></span>
<span id="cb18-2"><a href="data-summary.html#cb18-2" tabindex="-1"></a>    <span class="fu">filter</span>(species <span class="sc">==</span> <span class="st">&quot;s__&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb18-3"><a href="data-summary.html#cb18-3" tabindex="-1"></a>    <span class="fu">nrow</span>()</span></code></pre></div>
<pre class="script-output"><code>[1] 521</code></pre>
</div>
<div id="number-of-phylums" class="section level3 hasAnchor" number="2.3.1">
<h3><span class="header-section-number">2.3.1</span> Number of phylums<a href="data-summary.html#number-of-phylums" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb20-1"><a href="data-summary.html#cb20-1" tabindex="-1"></a>mags_table <span class="sc">%&gt;%</span></span>
<span id="cb20-2"><a href="data-summary.html#cb20-2" tabindex="-1"></a>    <span class="fu">select</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb20-3"><a href="data-summary.html#cb20-3" tabindex="-1"></a>  <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-4"><a href="data-summary.html#cb20-4" tabindex="-1"></a>  <span class="fu">pull</span>() <span class="sc">%&gt;%</span></span>
<span id="cb20-5"><a href="data-summary.html#cb20-5" tabindex="-1"></a>    <span class="fu">length</span>()</span></code></pre></div>
<pre class="script-output"><code>[1] 17</code></pre>
</div>
</div>
<div id="geographic-distribution" class="section level2 hasAnchor" number="2.4">
<h2><span class="header-section-number">2.4</span> Geographic distribution<a href="data-summary.html#geographic-distribution" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can visualise the origin of your samples using the geographic information available in the sample metadata table. First you need to generate summary information containing unique sampling sites and the number of samples per site.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb22-1"><a href="data-summary.html#cb22-1" tabindex="-1"></a><span class="co">#Summarise for generating map</span></span>
<span id="cb22-2"><a href="data-summary.html#cb22-2" tabindex="-1"></a><span class="fu">options</span>(<span class="at">dplyr.summarise.inform =</span> <span class="cn">FALSE</span>)</span>
<span id="cb22-3"><a href="data-summary.html#cb22-3" tabindex="-1"></a>sample_table_summary <span class="ot">&lt;-</span> sample_table <span class="sc">%&gt;%</span></span>
<span id="cb22-4"><a href="data-summary.html#cb22-4" tabindex="-1"></a>  <span class="co">#Group by geography and count samples</span></span>
<span id="cb22-5"><a href="data-summary.html#cb22-5" tabindex="-1"></a>  <span class="fu">select</span>(sample, latitude, longitude, country) <span class="sc">%&gt;%</span></span>
<span id="cb22-6"><a href="data-summary.html#cb22-6" tabindex="-1"></a>  <span class="fu">group_by</span>(latitude, longitude) <span class="sc">%&gt;%</span></span>
<span id="cb22-7"><a href="data-summary.html#cb22-7" tabindex="-1"></a>  <span class="fu">summarize</span>(<span class="at">count =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb22-8"><a href="data-summary.html#cb22-8" tabindex="-1"></a>  <span class="fu">ungroup</span>()</span></code></pre></div>
<p>Then, this new table can be used to generate the map with location shape sizes indicating the number of samples.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb23-1"><a href="data-summary.html#cb23-1" tabindex="-1"></a>sample_table_summary <span class="sc">%&gt;%</span></span>
<span id="cb23-2"><a href="data-summary.html#cb23-2" tabindex="-1"></a><span class="fu">ggplot</span>(.) <span class="sc">+</span></span>
<span id="cb23-3"><a href="data-summary.html#cb23-3" tabindex="-1"></a>    <span class="co">#render map</span></span>
<span id="cb23-4"><a href="data-summary.html#cb23-4" tabindex="-1"></a>    <span class="fu">geom_map</span>(</span>
<span id="cb23-5"><a href="data-summary.html#cb23-5" tabindex="-1"></a>      <span class="at">data=</span><span class="fu">map_data</span>(<span class="st">&quot;world&quot;</span>),</span>
<span id="cb23-6"><a href="data-summary.html#cb23-6" tabindex="-1"></a>      <span class="at">map =</span> <span class="fu">map_data</span>(<span class="st">&quot;world&quot;</span>),</span>
<span id="cb23-7"><a href="data-summary.html#cb23-7" tabindex="-1"></a>      <span class="fu">aes</span>(long, lat, <span class="at">map_id=</span>region),</span>
<span id="cb23-8"><a href="data-summary.html#cb23-8" tabindex="-1"></a>      <span class="at">color =</span> <span class="st">&quot;white&quot;</span>, <span class="at">fill =</span> <span class="st">&quot;#cccccc&quot;</span>, <span class="at">linewidth =</span> <span class="fl">0.2</span></span>
<span id="cb23-9"><a href="data-summary.html#cb23-9" tabindex="-1"></a>    ) <span class="sc">+</span></span>
<span id="cb23-10"><a href="data-summary.html#cb23-10" tabindex="-1"></a>    <span class="co">#render points</span></span>
<span id="cb23-11"><a href="data-summary.html#cb23-11" tabindex="-1"></a>    <span class="fu">geom_point</span>(</span>
<span id="cb23-12"><a href="data-summary.html#cb23-12" tabindex="-1"></a>      <span class="fu">aes</span>(<span class="at">x=</span>longitude,<span class="at">y=</span>latitude, <span class="at">size=</span>count),</span>
<span id="cb23-13"><a href="data-summary.html#cb23-13" tabindex="-1"></a>      <span class="at">alpha=</span><span class="fl">0.5</span>, <span class="at">shape=</span><span class="dv">16</span>) <span class="sc">+</span></span>
<span id="cb23-14"><a href="data-summary.html#cb23-14" tabindex="-1"></a>    <span class="co">#add general plot layout</span></span>
<span id="cb23-15"><a href="data-summary.html#cb23-15" tabindex="-1"></a>    <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb23-16"><a href="data-summary.html#cb23-16" tabindex="-1"></a>    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb23-17"><a href="data-summary.html#cb23-17" tabindex="-1"></a>        <span class="at">axis.title.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb23-18"><a href="data-summary.html#cb23-18" tabindex="-1"></a>        <span class="at">axis.title.y=</span><span class="fu">element_blank</span>()</span>
<span id="cb23-19"><a href="data-summary.html#cb23-19" tabindex="-1"></a>      )</span></code></pre></div>
<p><img src="_main_files/figure-html/geography_map-1.png" width="768" /></p>
<p>You can also plot the summary table.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb24-1"><a href="data-summary.html#cb24-1" tabindex="-1"></a>sample_table <span class="sc">%&gt;%</span></span>
<span id="cb24-2"><a href="data-summary.html#cb24-2" tabindex="-1"></a>  <span class="fu">select</span>(sample,sample_type,region,country,latitude,longitude) <span class="sc">%&gt;%</span></span>
<span id="cb24-3"><a href="data-summary.html#cb24-3" tabindex="-1"></a>  <span class="fu">kable</span>()</span></code></pre></div>
<table>
<colgroup>
<col width="13%" />
<col width="17%" />
<col width="19%" />
<col width="22%" />
<col width="13%" />
<col width="14%" />
</colgroup>
<thead>
<tr class="header">
<th align="left">sample</th>
<th align="left">sample_type</th>
<th align="left">region</th>
<th align="left">country</th>
<th align="right">latitude</th>
<th align="right">longitude</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">EHI00678</td>
<td align="left">Faecal</td>
<td align="left">San Salvador</td>
<td align="left">Bahamas</td>
<td align="right">24.05862</td>
<td align="right">-74.46739</td>
</tr>
<tr class="even">
<td align="left">EHI00682</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00743</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
<tr class="even">
<td align="left">EHI00679</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00674</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
<tr class="even">
<td align="left">EHI00689</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="odd">
<td align="left">EHI00688</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="even">
<td align="left">EHI00690</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="odd">
<td align="left">EHI00692</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="even">
<td align="left">EHI00753</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00696</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="even">
<td align="left">EHI00751</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
<tr class="odd">
<td align="left">EHI00680</td>
<td align="left">Faecal</td>
<td align="left">San Salvador</td>
<td align="left">Bahamas</td>
<td align="right">24.05862</td>
<td align="right">-74.46739</td>
</tr>
<tr class="even">
<td align="left">EHI00683</td>
<td align="left">Faecal</td>
<td align="left">San Salvador</td>
<td align="left">Bahamas</td>
<td align="right">24.05862</td>
<td align="right">-74.46739</td>
</tr>
<tr class="odd">
<td align="left">EHI00732</td>
<td align="left">Faecal</td>
<td align="left">Soroa</td>
<td align="left">Cuba</td>
<td align="right">22.77456</td>
<td align="right">-83.03659</td>
</tr>
<tr class="even">
<td align="left">EHI00695</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="odd">
<td align="left">EHI00693</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="even">
<td align="left">EHI00731</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="odd">
<td align="left">EHI00675</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
<tr class="even">
<td align="left">EHI00686</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="odd">
<td align="left">EHI00681</td>
<td align="left">Faecal</td>
<td align="left">San Salvador</td>
<td align="left">Bahamas</td>
<td align="right">24.05862</td>
<td align="right">-74.46739</td>
</tr>
<tr class="even">
<td align="left">EHI00758</td>
<td align="left">Faecal</td>
<td align="left">Soroa</td>
<td align="left">Cuba</td>
<td align="right">22.77456</td>
<td align="right">-83.03659</td>
</tr>
<tr class="odd">
<td align="left">EHI00685</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="even">
<td align="left">EHI00752</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00697</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="even">
<td align="left">EHI00742</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
<tr class="odd">
<td align="left">EHI00698</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="even">
<td align="left">EHI00687</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="odd">
<td align="left">EHI00684</td>
<td align="left">Faecal</td>
<td align="left">Acklins</td>
<td align="left">Bahamas</td>
<td align="right">22.65397</td>
<td align="right">-73.93493</td>
</tr>
<tr class="even">
<td align="left">EHI00733</td>
<td align="left">Faecal</td>
<td align="left">Soroa</td>
<td align="left">Cuba</td>
<td align="right">22.77456</td>
<td align="right">-83.03659</td>
</tr>
<tr class="odd">
<td align="left">EHI00691</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="even">
<td align="left">EHI00757</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00699</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="even">
<td align="left">EHI00677</td>
<td align="left">Faecal</td>
<td align="left">San Salvador</td>
<td align="left">Bahamas</td>
<td align="right">24.05862</td>
<td align="right">-74.46739</td>
</tr>
<tr class="odd">
<td align="left">EHI00700</td>
<td align="left">Faecal</td>
<td align="left">Grand Cayman</td>
<td align="left">Cayman Islands</td>
<td align="right">19.32504</td>
<td align="right">-81.20698</td>
</tr>
<tr class="even">
<td align="left">EHI00730</td>
<td align="left">Faecal</td>
<td align="left">Chub Cay</td>
<td align="left">Bahamas</td>
<td align="right">25.40986</td>
<td align="right">-77.87054</td>
</tr>
<tr class="odd">
<td align="left">EHI00694</td>
<td align="left">Faecal</td>
<td align="left">Andros</td>
<td align="left">Bahamas</td>
<td align="right">24.66547</td>
<td align="right">-77.80168</td>
</tr>
<tr class="even">
<td align="left">EHI00676</td>
<td align="left">Faecal</td>
<td align="left">Cockpit</td>
<td align="left">Jamaica</td>
<td align="right">18.38248</td>
<td align="right">-77.51726</td>
</tr>
</tbody>
</table>
</div>
<div id="prepare-the-colour-layout" class="section level2 hasAnchor" number="2.5">
<h2><span class="header-section-number">2.5</span> Prepare the colour layout<a href="data-summary.html#prepare-the-colour-layout" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>For the sake of consistency through EHI projects, we have created a unified colour profile for bacterial and archaeal taxa. The palette assign similar colours to closely related phyla, and more distinct hues to distantly related phyla. The colour profile can be downloaded from Github, and store as an object in R for downstream analyses.</p>
<p><a href="https://github.com/earthhologenome/EHI_taxonomy_colour/edit/main/README.md" class="uri">https://github.com/earthhologenome/EHI_taxonomy_colour/edit/main/README.md</a></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb25-1"><a href="data-summary.html#cb25-1" tabindex="-1"></a><span class="co"># Download and load the phylum colour table</span></span>
<span id="cb25-2"><a href="data-summary.html#cb25-2" tabindex="-1"></a>colours_URL<span class="ot">=</span><span class="st">&quot;https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/ehi_phylum_colors.tsv&quot;</span></span>
<span id="cb25-3"><a href="data-summary.html#cb25-3" tabindex="-1"></a><span class="fu">download.file</span>(colours_URL, <span class="st">&quot;data/ehi_phylum_colors.tsv&quot;</span>)</span>
<span id="cb25-4"><a href="data-summary.html#cb25-4" tabindex="-1"></a>ehi_phylum_colors <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">&quot;data/ehi_phylum_colors.tsv&quot;</span>,<span class="at">sep=</span><span class="st">&quot;</span><span class="sc">\t</span><span class="st">&quot;</span>,<span class="at">header=</span>T,<span class="at">comment.char =</span> <span class="st">&quot;&quot;</span>)</span>
<span id="cb25-5"><a href="data-summary.html#cb25-5" tabindex="-1"></a></span>
<span id="cb25-6"><a href="data-summary.html#cb25-6" tabindex="-1"></a><span class="co"># Arrange colors alphabetically</span></span>
<span id="cb25-7"><a href="data-summary.html#cb25-7" tabindex="-1"></a>colors_alphabetic <span class="ot">&lt;-</span> ehi_phylum_colors <span class="sc">%&gt;%</span></span>
<span id="cb25-8"><a href="data-summary.html#cb25-8" tabindex="-1"></a>  <span class="fu">right_join</span>(mags_table, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb25-9"><a href="data-summary.html#cb25-9" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb25-10"><a href="data-summary.html#cb25-10" tabindex="-1"></a>  <span class="fu">select</span>(phylum, colors) <span class="sc">%&gt;%</span></span>
<span id="cb25-11"><a href="data-summary.html#cb25-11" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb25-12"><a href="data-summary.html#cb25-12" tabindex="-1"></a>    <span class="fu">arrange</span>(phylum) <span class="sc">%&gt;%</span></span>
<span id="cb25-13"><a href="data-summary.html#cb25-13" tabindex="-1"></a>    <span class="fu">select</span>(colors) <span class="sc">%&gt;%</span></span>
<span id="cb25-14"><a href="data-summary.html#cb25-14" tabindex="-1"></a>    <span class="fu">pull</span>()</span></code></pre></div>
<div class="float">
<img src="https://raw.githubusercontent.com/earthhologenome/EHI_taxonomy_colour/main/phylum_tree.png" alt="Coloured GTDB 214 phylum tree" />
<div class="figcaption">Coloured GTDB 214 phylum tree</div>
</div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="mag-catalogue.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
