<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>The Earth Hologenome Initiative Data Analysis Workflow</title>
  <meta name="description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="The Earth Hologenome Initiative Data Analysis Workflow" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />
  <meta property="og:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="github-repo" content="earthhologenome/EHI_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="The Earth Hologenome Initiative Data Analysis Workflow" />
  
  <meta name="twitter:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="twitter:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />

<meta name="author" content="Ostaizka Aizpurua" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2023-11-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  

<link rel="next" href="data-summary.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><img src="images/ehi_logo.png" alt="Earth Hologenome Initiative"></li>
<li align="center"><b>ANALYSIS WORFLOW</b></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#download-mock-data"><i class="fa fa-check"></i><b>1.2</b> Download mock data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-summary.html"><a href="data-summary.html"><i class="fa fa-check"></i><b>2</b> Data summary</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-summary.html"><a href="data-summary.html#load-the-data"><i class="fa fa-check"></i><b>2.1</b> Load the data</a></li>
<li class="chapter" data-level="2.2" data-path="data-summary.html"><a href="data-summary.html#general-data-statistics"><i class="fa fa-check"></i><b>2.2</b> General data statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-samples"><i class="fa fa-check"></i>Number of samples</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-discarded-data"><i class="fa fa-check"></i>Amount of discarded data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-host-data"><i class="fa fa-check"></i>Amount of host data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-metagenomic-data"><i class="fa fa-check"></i>Amount of metagenomic data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-estimated-prokaryotic"><i class="fa fa-check"></i>Amount of estimated prokaryotic</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-summary.html"><a href="data-summary.html#general-mag-statistics"><i class="fa fa-check"></i><b>2.3</b> General MAG statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags"><i class="fa fa-check"></i>Number of MAGs</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags-without-species-level-annotation"><i class="fa fa-check"></i>Number of MAGs without species-level annotation</a></li>
<li class="chapter" data-level="2.3.1" data-path="data-summary.html"><a href="data-summary.html#number-of-phylums"><i class="fa fa-check"></i><b>2.3.1</b> Number of phylums</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-summary.html"><a href="data-summary.html#geographic-distribution"><i class="fa fa-check"></i><b>2.4</b> Geographic distribution</a></li>
<li class="chapter" data-level="2.5" data-path="data-summary.html"><a href="data-summary.html#prepare-the-colour-layout"><i class="fa fa-check"></i><b>2.5</b> Prepare the colour layout</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>3</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="3.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-phylogeny"><i class="fa fa-check"></i><b>3.1</b> MAG phylogeny</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-mag-metadata"><i class="fa fa-check"></i>Prepare MAG metadata</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylogenetic-tree"><i class="fa fa-check"></i>Plot phylogenetic tree</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylum-color-legend"><i class="fa fa-check"></i>Plot phylum color legend</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-quality"><i class="fa fa-check"></i><b>3.2</b> MAG quality</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-input-table"><i class="fa fa-check"></i>Prepare input table</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-biplot-chart"><i class="fa fa-check"></i>Create biplot chart</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-boxplot-charts"><i class="fa fa-check"></i>Create boxplot charts</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-composite-figure"><i class="fa fa-check"></i>Create composite figure</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-attributes-of-mags"><i class="fa fa-check"></i><b>3.3</b> Functional attributes of MAGs</a></li>
<li class="chapter" data-level="3.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-ordination-of-mags"><i class="fa fa-check"></i><b>3.4</b> Functional ordination of MAGs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html"><i class="fa fa-check"></i><b>4</b> Sequencing Assessment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#dna-fractions"><i class="fa fa-check"></i><b>4.1</b> DNA fractions</a></li>
<li class="chapter" data-level="4.2" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#estimated-vs.-mapped-prokaryotic-fraction"><i class="fa fa-check"></i><b>4.2</b> Estimated vs. mapped prokaryotic fraction</a></li>
<li class="chapter" data-level="4.3" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#additional-sequencing-needed"><i class="fa fa-check"></i><b>4.3</b> Additional sequencing needed</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="count-data.html"><a href="count-data.html"><i class="fa fa-check"></i><b>5</b> Count data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-data.html"><a href="count-data.html#minimum-coverage-filtering"><i class="fa fa-check"></i><b>5.1</b> Minimum coverage filtering</a></li>
<li class="chapter" data-level="5.2" data-path="count-data.html"><a href="count-data.html#genome-size-normalisation"><i class="fa fa-check"></i><b>5.2</b> Genome size normalisation</a></li>
<li class="chapter" data-level="5.3" data-path="count-data.html"><a href="count-data.html#count-table"><i class="fa fa-check"></i><b>5.3</b> Count table</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html"><i class="fa fa-check"></i><b>6</b> Taxonomic composition</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html#taxonomy-stacked-barplot"><i class="fa fa-check"></i><b>6.1</b> Taxonomy stacked barplot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> Diversity analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#data-preparation"><i class="fa fa-check"></i><b>7.1</b> Data preparation</a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity"><i class="fa fa-check"></i><b>7.2</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-metrics"><i class="fa fa-check"></i>Alpha diversity metrics</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-plots"><i class="fa fa-check"></i>Alpha diversity plots</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-comparisons"><i class="fa fa-check"></i>Alpha diversity comparisons</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#relationship-between-alpha-diversity-and-sequencing-effort"><i class="fa fa-check"></i>Relationship between alpha diversity and sequencing effort</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity"><i class="fa fa-check"></i><b>7.3</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-test"><i class="fa fa-check"></i>Beta diversity test</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-plot"><i class="fa fa-check"></i>Beta diversity plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://www.earthhologenome.org" target="blank"><b>Earth Hologenome Initiative</b></a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Earth Hologenome Initiative Data Analysis Workflow</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="header">
<h1 class="title">The Earth Hologenome Initiative Data Analysis Workflow</h1>
<p class="author"><em>Ostaizka Aizpurua<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a></em></p>
<p class="author"><em>Antton Alberdi<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></em></p>
<p class="date"><em>2023-11-03</em></p>
</div>
<div id="introduction" class="section level1 hasAnchor" number="1">
<h1><span class="header-section-number">Chapter 1</span> Introduction<a href="index.html#introduction" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The Earth Hologenome Initiative (EHI, <a href="http://www.earthhologenome.org">www.earthhologenome.org</a>) is a global collaborative endeavour aimed at promoting, facilitating, coordinating, and standardising hologenomic research on wild organisms worldwide. The EHI encompasses projects with diverse study designs and goals around standardised and open access sample collection and preservation, data generation and data management criteria.</p>
<p>One of the main objectives of the EHI is to facilitate analysis of animal genomic and microbial metagenomic data. Here, you will find resources to summarise, visualise and model data generated through the EHI pipeline.</p>
<div id="prepare-the-r-environment" class="section level2 hasAnchor" number="1.1">
<h2><span class="header-section-number">1.1</span> Prepare the R environment<a href="index.html#prepare-the-r-environment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The first step is to prepare your R environment. You must ensure you are at the appropriate working directory, and all required libraries are installed and loaded.</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb1-1"><a href="index.html#cb1-1" tabindex="-1"></a><span class="fu">setwd</span>(<span class="st">&quot;path_to_your_working_directory&quot;</span>) <span class="co"># Change this to your absolute working directory</span></span>
<span id="cb1-2"><a href="index.html#cb1-2" tabindex="-1"></a><span class="fu">dir_create</span>(<span class="st">&quot;data&quot;</span>) <span class="co">#Create the data folder to store EHI data files</span></span></code></pre></div>
<p>The pipeline requires the following packages to be installed in your R environment.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb2-1"><a href="index.html#cb2-1" tabindex="-1"></a><span class="co"># These packages can be installed from CRAN</span></span>
<span id="cb2-2"><a href="index.html#cb2-2" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;R.utils&quot;</span>)</span>
<span id="cb2-3"><a href="index.html#cb2-3" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;knitr&quot;</span>)</span>
<span id="cb2-4"><a href="index.html#cb2-4" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;tidyverse&quot;</span>)</span>
<span id="cb2-5"><a href="index.html#cb2-5" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ape&quot;</span>)</span>
<span id="cb2-6"><a href="index.html#cb2-6" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;phytools&quot;</span>)</span>
<span id="cb2-7"><a href="index.html#cb2-7" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggplot2&quot;</span>)</span>
<span id="cb2-8"><a href="index.html#cb2-8" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggtree&quot;</span>)</span>
<span id="cb2-9"><a href="index.html#cb2-9" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggrepel&quot;</span>)</span>
<span id="cb2-10"><a href="index.html#cb2-10" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggpubr&quot;</span>)</span>
<span id="cb2-11"><a href="index.html#cb2-11" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggnewscale&quot;</span>)</span>
<span id="cb2-12"><a href="index.html#cb2-12" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;ggtreeExtra&quot;</span>)</span>
<span id="cb2-13"><a href="index.html#cb2-13" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;gridExtra&quot;</span>)</span>
<span id="cb2-14"><a href="index.html#cb2-14" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;spaa&quot;</span>)</span>
<span id="cb2-15"><a href="index.html#cb2-15" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;vegan&quot;</span>)</span>
<span id="cb2-16"><a href="index.html#cb2-16" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;Rtsne&quot;</span>)</span>
<span id="cb2-17"><a href="index.html#cb2-17" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="st">&quot;devtools&quot;</span>)</span>
<span id="cb2-18"><a href="index.html#cb2-18" tabindex="-1"></a></span>
<span id="cb2-19"><a href="index.html#cb2-19" tabindex="-1"></a><span class="co"># This packages must be installed from Github using devtools</span></span>
<span id="cb2-20"><a href="index.html#cb2-20" tabindex="-1"></a><span class="fu">library</span>(devtools)</span>
<span id="cb2-21"><a href="index.html#cb2-21" tabindex="-1"></a><span class="fu">install_github</span>(<span class="st">&quot;anttonalberdi/hilldiv2&quot;</span>)</span></code></pre></div>
<p>All those packages then need to be loaded to your R environment.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb3-1"><a href="index.html#cb3-1" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(R.utils))</span>
<span id="cb3-2"><a href="index.html#cb3-2" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(knitr))</span>
<span id="cb3-3"><a href="index.html#cb3-3" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(tidyverse))</span>
<span id="cb3-4"><a href="index.html#cb3-4" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ape))</span>
<span id="cb3-5"><a href="index.html#cb3-5" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(phytools))</span>
<span id="cb3-6"><a href="index.html#cb3-6" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggplot2))</span>
<span id="cb3-7"><a href="index.html#cb3-7" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggtree))</span>
<span id="cb3-8"><a href="index.html#cb3-8" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggrepel))</span>
<span id="cb3-9"><a href="index.html#cb3-9" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggpubr))</span>
<span id="cb3-10"><a href="index.html#cb3-10" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggnewscale))</span>
<span id="cb3-11"><a href="index.html#cb3-11" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(ggtreeExtra))</span>
<span id="cb3-12"><a href="index.html#cb3-12" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(gridExtra))</span>
<span id="cb3-13"><a href="index.html#cb3-13" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(hilldiv2))</span>
<span id="cb3-14"><a href="index.html#cb3-14" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(spaa))</span>
<span id="cb3-15"><a href="index.html#cb3-15" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(vegan))</span>
<span id="cb3-16"><a href="index.html#cb3-16" tabindex="-1"></a><span class="fu">suppressPackageStartupMessages</span>(<span class="fu">library</span>(Rtsne))</span></code></pre></div>
</div>
<div id="download-mock-data" class="section level2 hasAnchor" number="1.2">
<h2><span class="header-section-number">1.2</span> Download mock data<a href="index.html#download-mock-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>If you want to reproduce the analyses shown in this website, you can download the mock data.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb4-1"><a href="index.html#cb4-1" tabindex="-1"></a><span class="co"># Count table</span></span>
<span id="cb4-2"><a href="index.html#cb4-2" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-3"><a href="index.html#cb4-3" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038_counts.tsv.gz&quot;</span>,</span>
<span id="cb4-4"><a href="index.html#cb4-4" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038_counts.tsv.gz&quot;</span>)</span>
<span id="cb4-5"><a href="index.html#cb4-5" tabindex="-1"></a></span>
<span id="cb4-6"><a href="index.html#cb4-6" tabindex="-1"></a><span class="co"># Coverage table</span></span>
<span id="cb4-7"><a href="index.html#cb4-7" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-8"><a href="index.html#cb4-8" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038_coverage.tsv.gz&quot;</span>,</span>
<span id="cb4-9"><a href="index.html#cb4-9" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038_coverage.tsv.gz&quot;</span>)</span>
<span id="cb4-10"><a href="index.html#cb4-10" tabindex="-1"></a></span>
<span id="cb4-11"><a href="index.html#cb4-11" tabindex="-1"></a><span class="co"># MAG metadata table</span></span>
<span id="cb4-12"><a href="index.html#cb4-12" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-13"><a href="index.html#cb4-13" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038_mag_info.tsv.gz&quot;</span>,</span>
<span id="cb4-14"><a href="index.html#cb4-14" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038_mag_info.tsv.gz&quot;</span>)</span>
<span id="cb4-15"><a href="index.html#cb4-15" tabindex="-1"></a></span>
<span id="cb4-16"><a href="index.html#cb4-16" tabindex="-1"></a><span class="co"># Sample metadata table</span></span>
<span id="cb4-17"><a href="index.html#cb4-17" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-18"><a href="index.html#cb4-18" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038_metadata.tsv.gz&quot;</span>,</span>
<span id="cb4-19"><a href="index.html#cb4-19" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038_metadata.tsv.gz&quot;</span>)</span>
<span id="cb4-20"><a href="index.html#cb4-20" tabindex="-1"></a></span>
<span id="cb4-21"><a href="index.html#cb4-21" tabindex="-1"></a><span class="co"># MAG phylogenetic tree</span></span>
<span id="cb4-22"><a href="index.html#cb4-22" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-23"><a href="index.html#cb4-23" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038.tree.gz&quot;</span>,</span>
<span id="cb4-24"><a href="index.html#cb4-24" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038.tree.gz&quot;</span>)</span>
<span id="cb4-25"><a href="index.html#cb4-25" tabindex="-1"></a></span>
<span id="cb4-26"><a href="index.html#cb4-26" tabindex="-1"></a><span class="co"># MAG functional annotation table</span></span>
<span id="cb4-27"><a href="index.html#cb4-27" tabindex="-1"></a><span class="fu">download.file</span>(</span>
<span id="cb4-28"><a href="index.html#cb4-28" tabindex="-1"></a>    <span class="st">&quot;https://github.com/earthhologenome/EHI_analysis/raw/main/data/DMB0038_merged_kegg.tsv.gz&quot;</span>,</span>
<span id="cb4-29"><a href="index.html#cb4-29" tabindex="-1"></a>    <span class="st">&quot;data/DMB0038_merged_kegg.tsv.gz&quot;</span>)</span></code></pre></div>
<p>Now you are ready to begin your analysis.</p>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="1">
<li id="fn1"><p>University of Copenhagen, <a href="mailto:ostaizka.aizpurua@sund.ku.dk" class="email">ostaizka.aizpurua@sund.ku.dk</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>University of Copenhagen, <a href="mailto:antton.alberdu@sund.ku.dk" class="email">antton.alberdu@sund.ku.dk</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>

<a href="data-summary.html" class="navigation navigation-next navigation-unique" aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
