<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>Chapter 3 MAG catalogue | The Earth Hologenome Initiative Data Analysis Workflow</title>
  <meta name="description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="Chapter 3 MAG catalogue | The Earth Hologenome Initiative Data Analysis Workflow" />
  <meta property="og:type" content="book" />
  <meta property="og:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />
  <meta property="og:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="github-repo" content="earthhologenome/EHI_analysis" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 MAG catalogue | The Earth Hologenome Initiative Data Analysis Workflow" />
  
  <meta name="twitter:description" content="Detailed information of the data analysis procedures employed in the Earth Hologenome Initiative." />
  <meta name="twitter:image" content="http://www.earthhologenome.org/analysis/images/ehi_logo.png" />

<meta name="author" content="Ostaizka Aizpurua" />
<meta name="author" content="Antton Alberdi" />


<meta name="date" content="2023-11-03" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="data-summary.html"/>
<link rel="next" href="sequencing-assessment.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="toc-logo"><img src="images/ehi_logo.png" alt="Earth Hologenome Initiative"></li>
<li align="center"><b>ANALYSIS WORFLOW</b></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Introduction</a>
<ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#prepare-the-r-environment"><i class="fa fa-check"></i><b>1.1</b> Prepare the R environment</a></li>
<li class="chapter" data-level="1.2" data-path="index.html"><a href="index.html#download-mock-data"><i class="fa fa-check"></i><b>1.2</b> Download mock data</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="data-summary.html"><a href="data-summary.html"><i class="fa fa-check"></i><b>2</b> Data summary</a>
<ul>
<li class="chapter" data-level="2.1" data-path="data-summary.html"><a href="data-summary.html#load-the-data"><i class="fa fa-check"></i><b>2.1</b> Load the data</a></li>
<li class="chapter" data-level="2.2" data-path="data-summary.html"><a href="data-summary.html#general-data-statistics"><i class="fa fa-check"></i><b>2.2</b> General data statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-samples"><i class="fa fa-check"></i>Number of samples</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-discarded-data"><i class="fa fa-check"></i>Amount of discarded data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-host-data"><i class="fa fa-check"></i>Amount of host data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-metagenomic-data"><i class="fa fa-check"></i>Amount of metagenomic data</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#amount-of-estimated-prokaryotic"><i class="fa fa-check"></i>Amount of estimated prokaryotic</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="data-summary.html"><a href="data-summary.html#general-mag-statistics"><i class="fa fa-check"></i><b>2.3</b> General MAG statistics</a>
<ul>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags"><i class="fa fa-check"></i>Number of MAGs</a></li>
<li class="chapter" data-level="" data-path="data-summary.html"><a href="data-summary.html#number-of-mags-without-species-level-annotation"><i class="fa fa-check"></i>Number of MAGs without species-level annotation</a></li>
<li class="chapter" data-level="2.3.1" data-path="data-summary.html"><a href="data-summary.html#number-of-phylums"><i class="fa fa-check"></i><b>2.3.1</b> Number of phylums</a></li>
</ul></li>
<li class="chapter" data-level="2.4" data-path="data-summary.html"><a href="data-summary.html#geographic-distribution"><i class="fa fa-check"></i><b>2.4</b> Geographic distribution</a></li>
<li class="chapter" data-level="2.5" data-path="data-summary.html"><a href="data-summary.html#prepare-the-colour-layout"><i class="fa fa-check"></i><b>2.5</b> Prepare the colour layout</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="mag-catalogue.html"><a href="mag-catalogue.html"><i class="fa fa-check"></i><b>3</b> MAG catalogue</a>
<ul>
<li class="chapter" data-level="3.1" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-phylogeny"><i class="fa fa-check"></i><b>3.1</b> MAG phylogeny</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-mag-metadata"><i class="fa fa-check"></i>Prepare MAG metadata</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylogenetic-tree"><i class="fa fa-check"></i>Plot phylogenetic tree</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#plot-phylum-color-legend"><i class="fa fa-check"></i>Plot phylum color legend</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="mag-catalogue.html"><a href="mag-catalogue.html#mag-quality"><i class="fa fa-check"></i><b>3.2</b> MAG quality</a>
<ul>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#prepare-input-table"><i class="fa fa-check"></i>Prepare input table</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-biplot-chart"><i class="fa fa-check"></i>Create biplot chart</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-boxplot-charts"><i class="fa fa-check"></i>Create boxplot charts</a></li>
<li class="chapter" data-level="" data-path="mag-catalogue.html"><a href="mag-catalogue.html#create-composite-figure"><i class="fa fa-check"></i>Create composite figure</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-attributes-of-mags"><i class="fa fa-check"></i><b>3.3</b> Functional attributes of MAGs</a></li>
<li class="chapter" data-level="3.4" data-path="mag-catalogue.html"><a href="mag-catalogue.html#functional-ordination-of-mags"><i class="fa fa-check"></i><b>3.4</b> Functional ordination of MAGs</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html"><i class="fa fa-check"></i><b>4</b> Sequencing Assessment</a>
<ul>
<li class="chapter" data-level="4.1" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#dna-fractions"><i class="fa fa-check"></i><b>4.1</b> DNA fractions</a></li>
<li class="chapter" data-level="4.2" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#estimated-vs.-mapped-prokaryotic-fraction"><i class="fa fa-check"></i><b>4.2</b> Estimated vs.Â mapped prokaryotic fraction</a></li>
<li class="chapter" data-level="4.3" data-path="sequencing-assessment.html"><a href="sequencing-assessment.html#additional-sequencing-needed"><i class="fa fa-check"></i><b>4.3</b> Additional sequencing needed</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="count-data.html"><a href="count-data.html"><i class="fa fa-check"></i><b>5</b> Count data</a>
<ul>
<li class="chapter" data-level="5.1" data-path="count-data.html"><a href="count-data.html#minimum-coverage-filtering"><i class="fa fa-check"></i><b>5.1</b> Minimum coverage filtering</a></li>
<li class="chapter" data-level="5.2" data-path="count-data.html"><a href="count-data.html#genome-size-normalisation"><i class="fa fa-check"></i><b>5.2</b> Genome size normalisation</a></li>
<li class="chapter" data-level="5.3" data-path="count-data.html"><a href="count-data.html#count-table"><i class="fa fa-check"></i><b>5.3</b> Count table</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html"><i class="fa fa-check"></i><b>6</b> Taxonomic composition</a>
<ul>
<li class="chapter" data-level="6.1" data-path="taxonomic-composition.html"><a href="taxonomic-composition.html#taxonomy-stacked-barplot"><i class="fa fa-check"></i><b>6.1</b> Taxonomy stacked barplot</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="diversity-analyses.html"><a href="diversity-analyses.html"><i class="fa fa-check"></i><b>7</b> Diversity analyses</a>
<ul>
<li class="chapter" data-level="7.1" data-path="diversity-analyses.html"><a href="diversity-analyses.html#data-preparation"><i class="fa fa-check"></i><b>7.1</b> Data preparation</a></li>
<li class="chapter" data-level="7.2" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity"><i class="fa fa-check"></i><b>7.2</b> Alpha diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-metrics"><i class="fa fa-check"></i>Alpha diversity metrics</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-plots"><i class="fa fa-check"></i>Alpha diversity plots</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#alpha-diversity-comparisons"><i class="fa fa-check"></i>Alpha diversity comparisons</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#relationship-between-alpha-diversity-and-sequencing-effort"><i class="fa fa-check"></i>Relationship between alpha diversity and sequencing effort</a></li>
</ul></li>
<li class="chapter" data-level="7.3" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity"><i class="fa fa-check"></i><b>7.3</b> Beta diversity</a>
<ul>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-test"><i class="fa fa-check"></i>Beta diversity test</a></li>
<li class="chapter" data-level="" data-path="diversity-analyses.html"><a href="diversity-analyses.html#beta-diversity-plot"><i class="fa fa-check"></i>Beta diversity plot</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="divider"></li>
<li><a href="https://www.earthhologenome.org" target="blank"><b>Earth Hologenome Initiative</b></a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">The Earth Hologenome Initiative Data Analysis Workflow</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="mag-catalogue" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">Chapter 3</span> MAG catalogue<a href="mag-catalogue.html#mag-catalogue" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>This workflow explores the characteristics of the MAG catalogue generated through the EHI pipeline. Note the workflow begins where the General analysis pipeline ends, so make sure you go through the general pipeline before starting working on this document.</p>
<div id="mag-phylogeny" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> MAG phylogeny<a href="mag-catalogue.html#mag-phylogeny" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can visualise the phylogeny of MAGs in multiple ways. Here we describe a workflow to generate a circular plot with a bunch of relevant information in outer rings.</p>
<div id="prepare-mag-metadata" class="section level3 unnumbered hasAnchor">
<h3>Prepare MAG metadata<a href="mag-catalogue.html#prepare-mag-metadata" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>First, we need to prepare the MAG metadata we want to plot in the outer rings of the tree.</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb26-1"><a href="mag-catalogue.html#cb26-1" tabindex="-1"></a><span class="co"># Which phylum the MAG belongs to</span></span>
<span id="cb26-2"><a href="mag-catalogue.html#cb26-2" tabindex="-1"></a>phyla <span class="ot">&lt;-</span> ehi_phylum_colors <span class="sc">%&gt;%</span></span>
<span id="cb26-3"><a href="mag-catalogue.html#cb26-3" tabindex="-1"></a>  <span class="fu">right_join</span>(mags_table, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb26-4"><a href="mag-catalogue.html#cb26-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb26-5"><a href="mag-catalogue.html#cb26-5" tabindex="-1"></a>  <span class="fu">select</span>(phylum, colors) <span class="sc">%&gt;%</span></span>
<span id="cb26-6"><a href="mag-catalogue.html#cb26-6" tabindex="-1"></a>    <span class="fu">unique</span>()</span>
<span id="cb26-7"><a href="mag-catalogue.html#cb26-7" tabindex="-1"></a></span>
<span id="cb26-8"><a href="mag-catalogue.html#cb26-8" tabindex="-1"></a><span class="co"># What is the genome size of the MAG in MBs (megabases)</span></span>
<span id="cb26-9"><a href="mag-catalogue.html#cb26-9" tabindex="-1"></a>mag_sizes <span class="ot">&lt;-</span> mags_table <span class="sc">%&gt;%</span></span>
<span id="cb26-10"><a href="mag-catalogue.html#cb26-10" tabindex="-1"></a>          <span class="fu">select</span>(<span class="fu">c</span>(genome,mag_size)) <span class="sc">%&gt;%</span></span>
<span id="cb26-11"><a href="mag-catalogue.html#cb26-11" tabindex="-1"></a>          <span class="fu">mutate</span>(<span class="at">mag_size=</span><span class="fu">round</span>(mag_size<span class="sc">/</span><span class="dv">1000000</span>,<span class="dv">2</span>))</span>
<span id="cb26-12"><a href="mag-catalogue.html#cb26-12" tabindex="-1"></a></span>
<span id="cb26-13"><a href="mag-catalogue.html#cb26-13" tabindex="-1"></a><span class="co"># What is the completeness of the MAG</span></span>
<span id="cb26-14"><a href="mag-catalogue.html#cb26-14" tabindex="-1"></a>mag_completeness <span class="ot">&lt;-</span> mags_table <span class="sc">%&gt;%</span></span>
<span id="cb26-15"><a href="mag-catalogue.html#cb26-15" tabindex="-1"></a>         <span class="fu">select</span>(<span class="fu">c</span>(genome,completeness)) <span class="sc">%&gt;%</span></span>
<span id="cb26-16"><a href="mag-catalogue.html#cb26-16" tabindex="-1"></a>         <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-17"><a href="mag-catalogue.html#cb26-17" tabindex="-1"></a>         <span class="fu">remove_rownames</span>() <span class="sc">%&gt;%</span></span>
<span id="cb26-18"><a href="mag-catalogue.html#cb26-18" tabindex="-1"></a>         <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>)</span>
<span id="cb26-19"><a href="mag-catalogue.html#cb26-19" tabindex="-1"></a></span>
<span id="cb26-20"><a href="mag-catalogue.html#cb26-20" tabindex="-1"></a><span class="co"># Generate the phylum color heatmap</span></span>
<span id="cb26-21"><a href="mag-catalogue.html#cb26-21" tabindex="-1"></a>heatmap <span class="ot">&lt;-</span> ehi_phylum_colors <span class="sc">%&gt;%</span></span>
<span id="cb26-22"><a href="mag-catalogue.html#cb26-22" tabindex="-1"></a>  <span class="fu">right_join</span>(mags_table, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb26-23"><a href="mag-catalogue.html#cb26-23" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb26-24"><a href="mag-catalogue.html#cb26-24" tabindex="-1"></a>  <span class="fu">select</span>(genome,phylum) <span class="sc">%&gt;%</span></span>
<span id="cb26-25"><a href="mag-catalogue.html#cb26-25" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> <span class="fu">unique</span>(phylum))) <span class="sc">%&gt;%</span></span>
<span id="cb26-26"><a href="mag-catalogue.html#cb26-26" tabindex="-1"></a>    <span class="fu">column_to_rownames</span>(<span class="at">var =</span> <span class="st">&quot;genome&quot;</span>)</span></code></pre></div>
</div>
<div id="plot-phylogenetic-tree" class="section level3 unnumbered hasAnchor">
<h3>Plot phylogenetic tree<a href="mag-catalogue.html#plot-phylogenetic-tree" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The next step is to plot the actual tree.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb27-1"><a href="mag-catalogue.html#cb27-1" tabindex="-1"></a><span class="co"># Create baseline circular tree</span></span>
<span id="cb27-2"><a href="mag-catalogue.html#cb27-2" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> <span class="fu">force.ultrametric</span>(tree,<span class="at">method=</span><span class="st">&quot;extend&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb27-3"><a href="mag-catalogue.html#cb27-3" tabindex="-1"></a>    <span class="fu">ggtree</span>(., <span class="at">layout =</span> <span class="st">&#39;circular&#39;</span>, <span class="at">size =</span> <span class="fl">0.3</span>, <span class="at">angle=</span><span class="dv">45</span>)</span></code></pre></div>
<pre class="script-output"><code>***************************************************************
*                          Note:                              *
*    force.ultrametric does not include a formal method to    *
*    ultrametricize a tree &amp; should only be used to coerce    *
*   a phylogeny that fails is.ultramtric due to rounding --   *
*    not as a substitute for formal rate-smoothing methods.   *
***************************************************************</code></pre>
<div class="sourceCode" id="cb29"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb29-1"><a href="mag-catalogue.html#cb29-1" tabindex="-1"></a><span class="co"># Add phylum colors ring</span></span>
<span id="cb29-2"><a href="mag-catalogue.html#cb29-2" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(circular_tree, heatmap, <span class="at">offset=</span><span class="fl">0.85</span>, <span class="at">width=</span><span class="fl">0.1</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb29-3"><a href="mag-catalogue.html#cb29-3" tabindex="-1"></a>        <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>colors_alphabetic) <span class="sc">+</span></span>
<span id="cb29-4"><a href="mag-catalogue.html#cb29-4" tabindex="-1"></a>        <span class="fu">geom_tiplab2</span>(<span class="at">size=</span><span class="dv">1</span>, <span class="at">hjust=</span><span class="sc">-</span><span class="fl">0.1</span>) <span class="sc">+</span></span>
<span id="cb29-5"><a href="mag-catalogue.html#cb29-5" tabindex="-1"></a>        <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>, <span class="at">plot.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>), <span class="at">panel.margin =</span> <span class="fu">margin</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>))</span>
<span id="cb29-6"><a href="mag-catalogue.html#cb29-6" tabindex="-1"></a></span>
<span id="cb29-7"><a href="mag-catalogue.html#cb29-7" tabindex="-1"></a><span class="co"># Flush color scale to enable a new color scheme in the next ring</span></span>
<span id="cb29-8"><a href="mag-catalogue.html#cb29-8" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span> circular_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb29-9"><a href="mag-catalogue.html#cb29-9" tabindex="-1"></a></span>
<span id="cb29-10"><a href="mag-catalogue.html#cb29-10" tabindex="-1"></a><span class="co"># Add completeness ring</span></span>
<span id="cb29-11"><a href="mag-catalogue.html#cb29-11" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span>  circular_tree <span class="sc">+</span></span>
<span id="cb29-12"><a href="mag-catalogue.html#cb29-12" tabindex="-1"></a>        <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb29-13"><a href="mag-catalogue.html#cb29-13" tabindex="-1"></a>        <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#d1f4ba&quot;</span>, <span class="at">high =</span> <span class="st">&quot;#f4baba&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-14"><a href="mag-catalogue.html#cb29-14" tabindex="-1"></a>        <span class="fu">geom_fruit</span>(</span>
<span id="cb29-15"><a href="mag-catalogue.html#cb29-15" tabindex="-1"></a>                <span class="at">data=</span>mags_table,</span>
<span id="cb29-16"><a href="mag-catalogue.html#cb29-16" tabindex="-1"></a>                <span class="at">geom=</span>geom_bar,</span>
<span id="cb29-17"><a href="mag-catalogue.html#cb29-17" tabindex="-1"></a>                <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>completeness, <span class="at">y=</span>genome, <span class="at">fill=</span>contamination),</span>
<span id="cb29-18"><a href="mag-catalogue.html#cb29-18" tabindex="-1"></a>                <span class="at">offset =</span> <span class="fl">0.55</span>,</span>
<span id="cb29-19"><a href="mag-catalogue.html#cb29-19" tabindex="-1"></a>                <span class="at">orientation=</span><span class="st">&quot;y&quot;</span>,</span>
<span id="cb29-20"><a href="mag-catalogue.html#cb29-20" tabindex="-1"></a>              <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)</span>
<span id="cb29-21"><a href="mag-catalogue.html#cb29-21" tabindex="-1"></a></span>
<span id="cb29-22"><a href="mag-catalogue.html#cb29-22" tabindex="-1"></a><span class="co"># Add genome-size ring</span></span>
<span id="cb29-23"><a href="mag-catalogue.html#cb29-23" tabindex="-1"></a>circular_tree <span class="ot">&lt;-</span>  circular_tree <span class="sc">+</span></span>
<span id="cb29-24"><a href="mag-catalogue.html#cb29-24" tabindex="-1"></a>        <span class="fu">new_scale_fill</span>() <span class="sc">+</span></span>
<span id="cb29-25"><a href="mag-catalogue.html#cb29-25" tabindex="-1"></a>        <span class="fu">scale_fill_manual</span>(<span class="at">values =</span> <span class="st">&quot;#cccccc&quot;</span>) <span class="sc">+</span></span>
<span id="cb29-26"><a href="mag-catalogue.html#cb29-26" tabindex="-1"></a>        <span class="fu">geom_fruit</span>(</span>
<span id="cb29-27"><a href="mag-catalogue.html#cb29-27" tabindex="-1"></a>             <span class="at">data=</span>mag_sizes,</span>
<span id="cb29-28"><a href="mag-catalogue.html#cb29-28" tabindex="-1"></a>             <span class="at">geom=</span>geom_bar,</span>
<span id="cb29-29"><a href="mag-catalogue.html#cb29-29" tabindex="-1"></a>             <span class="at">mapping =</span> <span class="fu">aes</span>(<span class="at">x=</span>mag_size, <span class="at">y=</span>genome),</span>
<span id="cb29-30"><a href="mag-catalogue.html#cb29-30" tabindex="-1"></a>                 <span class="at">offset =</span> <span class="fl">0.05</span>,</span>
<span id="cb29-31"><a href="mag-catalogue.html#cb29-31" tabindex="-1"></a>                 <span class="at">orientation=</span><span class="st">&quot;y&quot;</span>,</span>
<span id="cb29-32"><a href="mag-catalogue.html#cb29-32" tabindex="-1"></a>         <span class="at">stat=</span><span class="st">&quot;identity&quot;</span>)</span>
<span id="cb29-33"><a href="mag-catalogue.html#cb29-33" tabindex="-1"></a></span>
<span id="cb29-34"><a href="mag-catalogue.html#cb29-34" tabindex="-1"></a><span class="co">#Plot circular tree</span></span>
<span id="cb29-35"><a href="mag-catalogue.html#cb29-35" tabindex="-1"></a>circular_tree</span></code></pre></div>
<p><img src="_main_files/figure-html/tree_plot-1.png" width="768" /></p>
</div>
<div id="plot-phylum-color-legend" class="section level3 unnumbered hasAnchor">
<h3>Plot phylum color legend<a href="mag-catalogue.html#plot-phylum-color-legend" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can also create a custom phylum color legend using the following code.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb30-1"><a href="mag-catalogue.html#cb30-1" tabindex="-1"></a><span class="co"># Create legend</span></span>
<span id="cb30-2"><a href="mag-catalogue.html#cb30-2" tabindex="-1"></a>phyla_legend <span class="ot">&lt;-</span> ehi_phylum_colors <span class="sc">%&gt;%</span></span>
<span id="cb30-3"><a href="mag-catalogue.html#cb30-3" tabindex="-1"></a>  <span class="fu">right_join</span>(mags_table, <span class="at">by=</span><span class="fu">join_by</span>(phylum <span class="sc">==</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb30-4"><a href="mag-catalogue.html#cb30-4" tabindex="-1"></a>    <span class="fu">arrange</span>(<span class="fu">match</span>(genome, tree<span class="sc">$</span>tip.label)) <span class="sc">%&gt;%</span></span>
<span id="cb30-5"><a href="mag-catalogue.html#cb30-5" tabindex="-1"></a>  <span class="fu">select</span>(phylum, colors) <span class="sc">%&gt;%</span></span>
<span id="cb30-6"><a href="mag-catalogue.html#cb30-6" tabindex="-1"></a>    <span class="fu">unique</span>() <span class="sc">%&gt;%</span></span>
<span id="cb30-7"><a href="mag-catalogue.html#cb30-7" tabindex="-1"></a>    <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">gsub</span>(<span class="st">&quot;p__&quot;</span>,<span class="st">&quot;&quot;</span>,phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb30-8"><a href="mag-catalogue.html#cb30-8" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">phylum =</span> <span class="fu">factor</span>(phylum, <span class="at">levels =</span> phylum)) <span class="sc">%&gt;%</span></span>
<span id="cb30-9"><a href="mag-catalogue.html#cb30-9" tabindex="-1"></a>    <span class="fu">ggplot</span>() <span class="sc">+</span></span>
<span id="cb30-10"><a href="mag-catalogue.html#cb30-10" tabindex="-1"></a>      <span class="fu">geom_blank</span>() <span class="sc">+</span></span>
<span id="cb30-11"><a href="mag-catalogue.html#cb30-11" tabindex="-1"></a>      <span class="fu">geom_rect</span>(<span class="fu">aes</span>(<span class="at">xmin =</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(phyla)) <span class="sc">-</span> <span class="fl">0.5</span>, <span class="at">xmax =</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(phyla)) <span class="sc">+</span> <span class="fl">0.5</span>, <span class="at">ymin =</span> <span class="fl">0.19</span>, <span class="at">ymax =</span> <span class="fl">0.2</span>, <span class="at">fill =</span> phylum)) <span class="sc">+</span></span>
<span id="cb30-12"><a href="mag-catalogue.html#cb30-12" tabindex="-1"></a>        <span class="fu">scale_fill_manual</span>(<span class="at">values=</span><span class="fu">rev</span>(phyla<span class="sc">$</span>colors)) <span class="sc">+</span></span>
<span id="cb30-13"><a href="mag-catalogue.html#cb30-13" tabindex="-1"></a>      <span class="fu">geom_text</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="dv">1</span><span class="sc">:</span>(<span class="fu">nrow</span>(phyla)), <span class="at">y =</span> <span class="fl">0.15</span>, <span class="at">label =</span> <span class="fu">rev</span>(phylum)), <span class="at">angle =</span> <span class="dv">90</span>, <span class="at">hjust =</span> <span class="dv">0</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb30-14"><a href="mag-catalogue.html#cb30-14" tabindex="-1"></a>      <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb30-15"><a href="mag-catalogue.html#cb30-15" tabindex="-1"></a>      <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span>
<span id="cb30-16"><a href="mag-catalogue.html#cb30-16" tabindex="-1"></a></span>
<span id="cb30-17"><a href="mag-catalogue.html#cb30-17" tabindex="-1"></a><span class="co"># Plot legend</span></span>
<span id="cb30-18"><a href="mag-catalogue.html#cb30-18" tabindex="-1"></a>phyla_legend</span></code></pre></div>
<p><img src="_main_files/figure-html/tree_plot_legend-1.png" width="768" /></p>
</div>
</div>
<div id="mag-quality" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> MAG quality<a href="mag-catalogue.html#mag-quality" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>You can next plot a 2-dimensional plot showing the two main parameters used to assess quality of reconstructed MAG: completeness and contamination (or redundancy).</p>
<p>These parameters are calculated by CheckM based on the presence of single-copy core genes in the MAGs. Completeness values should ideally be as close to 100 as possible, because incompleteness introduces biases when performing functional analyses. Contamination should be as close to 0 as possible, because it indicates that the MAG probably contains DNA fragments that in reality belong to a different genome, and therefore can distort the results and introduce noise.</p>
<div id="prepare-input-table" class="section level3 unnumbered hasAnchor">
<h3>Prepare input table<a href="mag-catalogue.html#prepare-input-table" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You first need to prepare the input table from the original MAG metadata table.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb31-1"><a href="mag-catalogue.html#cb31-1" tabindex="-1"></a>mag_details <span class="ot">&lt;-</span> mags_table <span class="sc">%&gt;%</span></span>
<span id="cb31-2"><a href="mag-catalogue.html#cb31-2" tabindex="-1"></a>  <span class="fu">select</span>(<span class="fu">c</span>(genome,domain,phylum,completeness,contamination,mag_size)) <span class="sc">%&gt;%</span></span>
<span id="cb31-3"><a href="mag-catalogue.html#cb31-3" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">mag_size=</span><span class="fu">round</span>(mag_size<span class="sc">/</span><span class="dv">1000000</span>,<span class="dv">2</span>)) <span class="sc">%&gt;%</span> <span class="co">#change mag_size to MBs</span></span>
<span id="cb31-4"><a href="mag-catalogue.html#cb31-4" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">comp=</span>completeness,<span class="at">cont=</span>contamination,<span class="at">size=</span>mag_size) <span class="sc">%&gt;%</span> <span class="co">#rename columns</span></span>
<span id="cb31-5"><a href="mag-catalogue.html#cb31-5" tabindex="-1"></a>  <span class="fu">remove_rownames</span>() <span class="sc">%&gt;%</span></span>
<span id="cb31-6"><a href="mag-catalogue.html#cb31-6" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">match</span>(genome, <span class="fu">rev</span>(tree<span class="sc">$</span>tip.label))) <span class="co">#sort MAGs according to phylogenetic tree</span></span></code></pre></div>
</div>
<div id="create-biplot-chart" class="section level3 unnumbered hasAnchor">
<h3>Create biplot chart<a href="mag-catalogue.html#create-biplot-chart" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Using the code below you can create a plot in which MAGs are ordinated according to their completeness and contamination scores, while dot sizes indicate their genome-size.</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb32-1"><a href="mag-catalogue.html#cb32-1" tabindex="-1"></a>mag_stats_biplot <span class="ot">&lt;-</span> mag_details <span class="sc">%&gt;%</span></span>
<span id="cb32-2"><a href="mag-catalogue.html#cb32-2" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>comp,<span class="at">y=</span>cont,<span class="at">size=</span>size,<span class="at">color=</span>phylum)) <span class="sc">+</span></span>
<span id="cb32-3"><a href="mag-catalogue.html#cb32-3" tabindex="-1"></a>              <span class="fu">geom_point</span>(<span class="at">alpha=</span><span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb32-4"><a href="mag-catalogue.html#cb32-4" tabindex="-1"></a>                    <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb32-5"><a href="mag-catalogue.html#cb32-5" tabindex="-1"></a>                    <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colors_alphabetic) <span class="sc">+</span></span>
<span id="cb32-6"><a href="mag-catalogue.html#cb32-6" tabindex="-1"></a>                <span class="fu">labs</span>(<span class="at">y=</span> <span class="st">&quot;Contamination&quot;</span>, <span class="at">x =</span> <span class="st">&quot;Completeness&quot;</span>) <span class="sc">+</span></span>
<span id="cb32-7"><a href="mag-catalogue.html#cb32-7" tabindex="-1"></a>                    <span class="fu">theme_classic</span>() <span class="sc">+</span></span>
<span id="cb32-8"><a href="mag-catalogue.html#cb32-8" tabindex="-1"></a>                  <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
</div>
<div id="create-boxplot-charts" class="section level3 unnumbered hasAnchor">
<h3>Create boxplot charts<a href="mag-catalogue.html#create-boxplot-charts" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>You can also create X and Y axis boxplots to complement the above plot and better visualise MAG quality statistics.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb33-1"><a href="mag-catalogue.html#cb33-1" tabindex="-1"></a>mag_stats_cont <span class="ot">&lt;-</span> mag_details <span class="sc">%&gt;%</span></span>
<span id="cb33-2"><a href="mag-catalogue.html#cb33-2" tabindex="-1"></a>            <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">y=</span>cont)) <span class="sc">+</span></span>
<span id="cb33-3"><a href="mag-catalogue.html#cb33-3" tabindex="-1"></a>                    <span class="fu">ylim</span>(<span class="fu">c</span>(<span class="dv">10</span>,<span class="dv">0</span>)) <span class="sc">+</span></span>
<span id="cb33-4"><a href="mag-catalogue.html#cb33-4" tabindex="-1"></a>                    <span class="fu">geom_boxplot</span>(<span class="at">colour =</span> <span class="st">&quot;#999999&quot;</span>, <span class="at">fill=</span><span class="st">&quot;#cccccc&quot;</span>) <span class="sc">+</span></span>
<span id="cb33-5"><a href="mag-catalogue.html#cb33-5" tabindex="-1"></a>                    <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb33-6"><a href="mag-catalogue.html#cb33-6" tabindex="-1"></a>                    <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb33-7"><a href="mag-catalogue.html#cb33-7" tabindex="-1"></a>                        <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-8"><a href="mag-catalogue.html#cb33-8" tabindex="-1"></a>                        <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb33-9"><a href="mag-catalogue.html#cb33-9" tabindex="-1"></a>                        <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb33-10"><a href="mag-catalogue.html#cb33-10" tabindex="-1"></a>                  <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb33-11"><a href="mag-catalogue.html#cb33-11" tabindex="-1"></a>                        <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb33-12"><a href="mag-catalogue.html#cb33-12" tabindex="-1"></a>                  <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb33-13"><a href="mag-catalogue.html#cb33-13" tabindex="-1"></a>                        <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.40</span>, <span class="dv">0</span>),<span class="st">&quot;inches&quot;</span>)) <span class="co">#add bottom-margin (top, right, bottom, left)</span></span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb34-1"><a href="mag-catalogue.html#cb34-1" tabindex="-1"></a>mag_stats_comp <span class="ot">&lt;-</span>mag_details <span class="sc">%&gt;%</span></span>
<span id="cb34-2"><a href="mag-catalogue.html#cb34-2" tabindex="-1"></a>        <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x=</span>comp)) <span class="sc">+</span></span>
<span id="cb34-3"><a href="mag-catalogue.html#cb34-3" tabindex="-1"></a>                <span class="fu">xlim</span>(<span class="fu">c</span>(<span class="dv">50</span>,<span class="dv">100</span>)) <span class="sc">+</span></span>
<span id="cb34-4"><a href="mag-catalogue.html#cb34-4" tabindex="-1"></a>                <span class="fu">geom_boxplot</span>(<span class="at">colour =</span> <span class="st">&quot;#999999&quot;</span>, <span class="at">fill=</span><span class="st">&quot;#cccccc&quot;</span>) <span class="sc">+</span></span>
<span id="cb34-5"><a href="mag-catalogue.html#cb34-5" tabindex="-1"></a>                <span class="fu">theme_void</span>() <span class="sc">+</span></span>
<span id="cb34-6"><a href="mag-catalogue.html#cb34-6" tabindex="-1"></a>                <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>,</span>
<span id="cb34-7"><a href="mag-catalogue.html#cb34-7" tabindex="-1"></a>                    <span class="at">axis.title.x =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-8"><a href="mag-catalogue.html#cb34-8" tabindex="-1"></a>                    <span class="at">axis.title.y =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb34-9"><a href="mag-catalogue.html#cb34-9" tabindex="-1"></a>                    <span class="at">axis.text.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb34-10"><a href="mag-catalogue.html#cb34-10" tabindex="-1"></a>              <span class="at">axis.ticks.y=</span><span class="fu">element_blank</span>(),</span>
<span id="cb34-11"><a href="mag-catalogue.html#cb34-11" tabindex="-1"></a>                    <span class="at">axis.text.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb34-12"><a href="mag-catalogue.html#cb34-12" tabindex="-1"></a>              <span class="at">axis.ticks.x=</span><span class="fu">element_blank</span>(),</span>
<span id="cb34-13"><a href="mag-catalogue.html#cb34-13" tabindex="-1"></a>                    <span class="at">plot.margin =</span> <span class="fu">unit</span>(<span class="fu">c</span>(<span class="dv">0</span>, <span class="dv">0</span>, <span class="dv">0</span>, <span class="fl">0.50</span>),<span class="st">&quot;inches&quot;</span>)) <span class="co">#add left-margin (top, right, bottom, left)</span></span></code></pre></div>
</div>
<div id="create-composite-figure" class="section level3 unnumbered hasAnchor">
<h3>Create composite figure<a href="mag-catalogue.html#create-composite-figure" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Finally, you can plot a composite figure containing all above charts. The complex layout matrix only defines the space of each plot in the composite figure.</p>
<div class="sourceCode" id="cb35"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb35-1"><a href="mag-catalogue.html#cb35-1" tabindex="-1"></a><span class="fu">grid.arrange</span>(<span class="at">grobs =</span> <span class="fu">list</span>(mag_stats_comp,mag_stats_biplot,mag_stats_cont),</span>
<span id="cb35-2"><a href="mag-catalogue.html#cb35-2" tabindex="-1"></a>        <span class="at">layout_matrix =</span> <span class="fu">rbind</span>(<span class="fu">c</span>(<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">1</span>,<span class="dv">4</span>),</span>
<span id="cb35-3"><a href="mag-catalogue.html#cb35-3" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-4"><a href="mag-catalogue.html#cb35-4" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-5"><a href="mag-catalogue.html#cb35-5" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-6"><a href="mag-catalogue.html#cb35-6" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-7"><a href="mag-catalogue.html#cb35-7" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-8"><a href="mag-catalogue.html#cb35-8" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-9"><a href="mag-catalogue.html#cb35-9" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-10"><a href="mag-catalogue.html#cb35-10" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-11"><a href="mag-catalogue.html#cb35-11" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-12"><a href="mag-catalogue.html#cb35-12" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>),</span>
<span id="cb35-13"><a href="mag-catalogue.html#cb35-13" tabindex="-1"></a>                                                    <span class="fu">c</span>(<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">2</span>,<span class="dv">3</span>)))</span></code></pre></div>
<p><img src="_main_files/figure-html/mag_quality_composite-1.png" width="768" /></p>
</div>
</div>
<div id="functional-attributes-of-mags" class="section level2 hasAnchor" number="3.3">
<h2><span class="header-section-number">3.3</span> Functional attributes of MAGs<a href="mag-catalogue.html#functional-attributes-of-mags" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Prokaryotic genomes carry different sets of genes that confer their organisms with different functional capabilities By annotating the microbial genes against the KEGG database, enables calculating how full each of the hundreds of metabolic functions is in each genome, and in this way infer functional capabilities of each MAG.</p>
<p>Using the code below, you can create a heatmap with an associated phylogenetic tree showing how the MAG phylogenetic tree is associated with fullness (between 0-1) levels for each metabolic pathway.</p>
<div class="sourceCode" id="cb36"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb36-1"><a href="mag-catalogue.html#cb36-1" tabindex="-1"></a><span class="co">#Generate a basal utrametric tree for the sake of visualisation</span></span>
<span id="cb36-2"><a href="mag-catalogue.html#cb36-2" tabindex="-1"></a>kegg_tree <span class="ot">&lt;-</span> <span class="fu">force.ultrametric</span>(tree,<span class="at">method=</span><span class="st">&quot;extend&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb36-3"><a href="mag-catalogue.html#cb36-3" tabindex="-1"></a>                <span class="fu">ggtree</span>(., <span class="at">size =</span> <span class="fl">0.3</span>)</span></code></pre></div>
<pre class="script-output"><code>***************************************************************
*                          Note:                              *
*    force.ultrametric does not include a formal method to    *
*    ultrametricize a tree &amp; should only be used to coerce    *
*   a phylogeny that fails is.ultramtric due to rounding --   *
*    not as a substitute for formal rate-smoothing methods.   *
***************************************************************</code></pre>
<div class="sourceCode" id="cb38"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb38-1"><a href="mag-catalogue.html#cb38-1" tabindex="-1"></a><span class="co">#Add phylum colors next to the tree tips</span></span>
<span id="cb38-2"><a href="mag-catalogue.html#cb38-2" tabindex="-1"></a>    kegg_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(kegg_tree, heatmap, <span class="at">offset=</span><span class="dv">0</span>, <span class="at">width=</span><span class="fl">0.1</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb38-3"><a href="mag-catalogue.html#cb38-3" tabindex="-1"></a>            <span class="fu">scale_fill_manual</span>(<span class="at">values=</span>colors_alphabetic)</span>
<span id="cb38-4"><a href="mag-catalogue.html#cb38-4" tabindex="-1"></a></span>
<span id="cb38-5"><a href="mag-catalogue.html#cb38-5" tabindex="-1"></a><span class="co">#Reset fill scale to use a different colour profile in the heatmap</span></span>
<span id="cb38-6"><a href="mag-catalogue.html#cb38-6" tabindex="-1"></a>kegg_tree <span class="ot">&lt;-</span> kegg_tree <span class="sc">+</span> <span class="fu">new_scale_fill</span>()</span>
<span id="cb38-7"><a href="mag-catalogue.html#cb38-7" tabindex="-1"></a></span>
<span id="cb38-8"><a href="mag-catalogue.html#cb38-8" tabindex="-1"></a><span class="co">#Add KEGG heatmap</span></span>
<span id="cb38-9"><a href="mag-catalogue.html#cb38-9" tabindex="-1"></a>kegg_tree <span class="ot">&lt;-</span> <span class="fu">gheatmap</span>(kegg_tree, kegg_table, <span class="at">offset=</span><span class="fl">0.5</span>, <span class="at">width=</span><span class="fl">3.5</span>, <span class="at">colnames=</span><span class="cn">FALSE</span>) <span class="sc">+</span></span>
<span id="cb38-10"><a href="mag-catalogue.html#cb38-10" tabindex="-1"></a>            <span class="fu">vexpand</span>(.<span class="dv">08</span>) <span class="sc">+</span></span>
<span id="cb38-11"><a href="mag-catalogue.html#cb38-11" tabindex="-1"></a>            <span class="fu">coord_cartesian</span>(<span class="at">clip =</span> <span class="st">&quot;off&quot;</span>) <span class="sc">+</span></span>
<span id="cb38-12"><a href="mag-catalogue.html#cb38-12" tabindex="-1"></a>            <span class="fu">scale_fill_gradient</span>(<span class="at">low =</span> <span class="st">&quot;#f4f4f4&quot;</span>, <span class="at">high =</span> <span class="st">&quot;steelblue&quot;</span>, <span class="at">na.value=</span><span class="st">&quot;white&quot;</span>)</span>
<span id="cb38-13"><a href="mag-catalogue.html#cb38-13" tabindex="-1"></a></span>
<span id="cb38-14"><a href="mag-catalogue.html#cb38-14" tabindex="-1"></a><span class="co">#Plot combined tree + heatmap</span></span>
<span id="cb38-15"><a href="mag-catalogue.html#cb38-15" tabindex="-1"></a>kegg_tree <span class="sc">+</span></span>
<span id="cb38-16"><a href="mag-catalogue.html#cb38-16" tabindex="-1"></a>            <span class="fu">theme</span>(<span class="at">legend.position=</span><span class="st">&#39;none&#39;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/mag_kegg_heatmap-1.png" width="768" /></p>
</div>
<div id="functional-ordination-of-mags" class="section level2 hasAnchor" number="3.4">
<h2><span class="header-section-number">3.4</span> Functional ordination of MAGs<a href="mag-catalogue.html#functional-ordination-of-mags" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using the functional annotation, it is possible to ordinate prokaryotic genomes on a bidimentional space. In doing so, one can assess how close any group of bacteria are in functional terms, or how functionally diverse the members of a given phylum can be. In this example the ordination is conducted using the tSNE method.</p>
<div class="sourceCode" id="cb39"><pre class="sourceCode r script-source"><code class="sourceCode r"><span id="cb39-1"><a href="mag-catalogue.html#cb39-1" tabindex="-1"></a><span class="co"># Generate the tSNE ordination</span></span>
<span id="cb39-2"><a href="mag-catalogue.html#cb39-2" tabindex="-1"></a>tSNE_func <span class="ot">&lt;-</span> <span class="fu">Rtsne</span>(<span class="at">X=</span>kegg_table, <span class="at">dims =</span> <span class="dv">2</span>, <span class="at">check_duplicates =</span> <span class="cn">FALSE</span>)</span>
<span id="cb39-3"><a href="mag-catalogue.html#cb39-3" tabindex="-1"></a></span>
<span id="cb39-4"><a href="mag-catalogue.html#cb39-4" tabindex="-1"></a><span class="co"># Plot the ordination</span></span>
<span id="cb39-5"><a href="mag-catalogue.html#cb39-5" tabindex="-1"></a>tSNE_func<span class="sc">$</span>Y <span class="sc">%&gt;%</span></span>
<span id="cb39-6"><a href="mag-catalogue.html#cb39-6" tabindex="-1"></a>                <span class="fu">as.data.frame</span>() <span class="sc">%&gt;%</span></span>
<span id="cb39-7"><a href="mag-catalogue.html#cb39-7" tabindex="-1"></a>                <span class="fu">mutate</span>(<span class="at">genome=</span><span class="fu">rownames</span>(kegg_table)) <span class="sc">%&gt;%</span></span>
<span id="cb39-8"><a href="mag-catalogue.html#cb39-8" tabindex="-1"></a>                <span class="fu">inner_join</span>(mags_table, <span class="at">by=</span><span class="st">&quot;genome&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-9"><a href="mag-catalogue.html#cb39-9" tabindex="-1"></a>                <span class="fu">rename</span>(<span class="at">tSNE1=</span><span class="st">&quot;V1&quot;</span>, <span class="at">tSNE2=</span><span class="st">&quot;V2&quot;</span>) <span class="sc">%&gt;%</span></span>
<span id="cb39-10"><a href="mag-catalogue.html#cb39-10" tabindex="-1"></a>                <span class="fu">select</span>(genome,phylum,tSNE1,tSNE2, completeness) <span class="sc">%&gt;%</span></span>
<span id="cb39-11"><a href="mag-catalogue.html#cb39-11" tabindex="-1"></a>                <span class="fu">ggplot</span>(<span class="fu">aes</span>(<span class="at">x =</span> tSNE1, <span class="at">y =</span> tSNE2, <span class="at">color =</span> phylum, <span class="at">size=</span>completeness))<span class="sc">+</span></span>
<span id="cb39-12"><a href="mag-catalogue.html#cb39-12" tabindex="-1"></a>                            <span class="fu">geom_point</span>(<span class="at">shape=</span><span class="dv">16</span>, <span class="at">alpha=</span><span class="fl">0.7</span>) <span class="sc">+</span></span>
<span id="cb39-13"><a href="mag-catalogue.html#cb39-13" tabindex="-1"></a>                            <span class="fu">scale_color_manual</span>(<span class="at">values=</span>colors_alphabetic) <span class="sc">+</span></span>
<span id="cb39-14"><a href="mag-catalogue.html#cb39-14" tabindex="-1"></a>                            <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb39-15"><a href="mag-catalogue.html#cb39-15" tabindex="-1"></a>                            <span class="fu">theme</span>(<span class="at">legend.position =</span> <span class="st">&quot;none&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/mag_kegg_ordination-1.png" width="768" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="data-summary.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="sequencing-assessment.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": ["_main.pdf", "_main.epub"],
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
